<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUChicago" />

<meta name="date" content="2021-01-14" />

<title>convert_tsv_to_dataframe</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BH_Diel</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/BH_Diel">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">convert_tsv_to_dataframe</h1>
<h4 class="author">KiseokUChicago</h4>
<h4 class="date">2021-01-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BH_Diel/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210222code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210222)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210222code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210222)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoBHDieltreefb4d76396edc94e7c43cff8cad6db42ebadbaabftargetblankfb4d763a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/BH_Diel/tree/fb4d76396edc94e7c43cff8cad6db42ebadbaabf" target="_blank">fb4d763</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoBHDieltreefb4d76396edc94e7c43cff8cad6db42ebadbaabftargetblankfb4d763a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/BH_Diel/tree/fb4d76396edc94e7c43cff8cad6db42ebadbaabf" target="_blank">fb4d763</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  01_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-BDCL_filtered_PSMs.tsv
    Untracked:  01_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  02_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-DCL_filtered_PSMs.tsv
    Untracked:  02_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-DCL_filtered_PSMs.xlsx
    Untracked:  03_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-SRF_filtered_PSMs.tsv
    Untracked:  03_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-SRF_filtered_PSMs.xlsx
    Untracked:  04_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-BDCL_filtered_PSMs.tsv
    Untracked:  04_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-BDCL_filtered_PSMs.xlsx
    Untracked:  05_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-DCL_filtered_PSMs.tsv
    Untracked:  05_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-DCL_filtered_PSMs.xlsx
    Untracked:  06_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-SRF_filtered_PSMs.tsv
    Untracked:  06_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-SRF_filtered_PSMs.xlsx
    Untracked:  07_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-BDCL_filtered_PSMs.tsv
    Untracked:  07_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  08_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-DCL_filtered_PSMs.tsv
    Untracked:  08_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-DCL_filtered_PSMs.xlsx
    Untracked:  09_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-SRF_filtered_PSMs.tsv
    Untracked:  09_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-SRF_filtered_PSMs.xlsx
    Untracked:  10_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-BDCL_filtered_PSMs.tsv
    Untracked:  10_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  11_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-DCL_filtered_PSMs.tsv
    Untracked:  11_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-DCL_filtered_PSMs.xlsx
    Untracked:  12_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-SRF_filtered_PSMs.tsv
    Untracked:  12_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-SRF_filtered_PSMs.xlsx
    Untracked:  13_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-BDCL_filtered_PSMs.tsv
    Untracked:  13_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  14_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-DCL_filtered_PSMs.tsv
    Untracked:  14_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-DCL_filtered_PSMs.xlsx
    Untracked:  15_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-SRF_filtered_PSMs.tsv
    Untracked:  15_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-SRF_filtered_PSMs.xlsx
    Untracked:  16_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-BDCL_filtered_PSMs.tsv
    Untracked:  16_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  17_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-DCL_filtered_PSMs.tsv
    Untracked:  17_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-DCL_filtered_PSMs.xlsx
    Untracked:  18_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-SRF_filtered_PSMs.tsv
    Untracked:  18_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-SRF_filtered_PSMs.xlsx
    Untracked:  19_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-DCL_filtered_PSMs.tsv
    Untracked:  19_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-DCL_filtered_PSMs.xlsx
    Untracked:  20_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-SRF_filtered_PSMs.tsv
    Untracked:  20_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-SRF_filtered_PSMs.xlsx
    Untracked:  21_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-BDCL_filtered_PSMs.tsv
    Untracked:  21_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  22_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-DCL_filtered_PSMs.tsv
    Untracked:  22_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-DCL_filtered_PSMs.xlsx
    Untracked:  23_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-SRF_filtered_PSMs.tsv
    Untracked:  23_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-SRF_filtered_PSMs.xlsx
    Untracked:  24_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.tsv
    Untracked:  24_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  25_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-DCL_filtered_PSMs.tsv
    Untracked:  25_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-DCL_filtered_PSMs.xlsx
    Untracked:  26_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-SRF_filtered_PSMs.tsv
    Untracked:  26_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-SRF_filtered_PSMs.xlsx
    Untracked:  27_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-BDCL_filtered_PSMs.tsv
    Untracked:  27_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-BDCL_filtered_PSMs.xlsx
    Untracked:  28_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-DCL_filtered_PSMs.tsv
    Untracked:  28_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-DCL_filtered_PSMs.xlsx
    Untracked:  29_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-SRF_filtered_PSMs.tsv
    Untracked:  29_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-SRF_filtered_PSMs.xlsx
    Untracked:  30_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-BDCL_filtered_PSMs.tsv
    Untracked:  30_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  31_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-DCL_filtered_PSMs.tsv
    Untracked:  31_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-DCL_filtered_PSMs.xlsx
    Untracked:  32_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-SRF_filtered_PSMs.tsv
    Untracked:  32_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-SRF_filtered_PSMs.xlsx
    Untracked:  33_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-BDCL_filtered_PSMs.tsv
    Untracked:  33_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  34_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-DCL_filtered_PSMs.tsv
    Untracked:  34_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-DCL_filtered_PSMs.xlsx
    Untracked:  35_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-SRF_filtered_PSMs.tsv
    Untracked:  35_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-SRF_filtered_PSMs.xlsx
    Untracked:  36_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-BDCL_filtered_PSMs.tsv
    Untracked:  36_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  37_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-DCL_filtered_PSMs.tsv
    Untracked:  37_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-DCL_filtered_PSMs.xlsx
    Untracked:  38_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-SRF_filtered_PSMs.tsv
    Untracked:  38_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-SRF_filtered_PSMs.xlsx
    Untracked:  39_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-BDCL_filtered_PSMs.tsv
    Untracked:  39_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  40_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-DCL_filtered_PSMs.tsv
    Untracked:  40_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-DCL_filtered_PSMs.xlsx
    Untracked:  41_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-SRF_filtered_PSMs.tsv
    Untracked:  41_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-SRF_filtered_PSMs.xlsx
    Untracked:  42_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-BDCL_filtered_PSMs.tsv
    Untracked:  42_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-BDCL_filtered_PSMs.xlsx
    Untracked:  43_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-DCL_filtered_PSMs.tsv
    Untracked:  43_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-DCL_filtered_PSMs.xlsx
    Untracked:  44_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-SRF_filtered_PSMs.tsv
    Untracked:  44_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-SRF_filtered_PSMs.xlsx
    Untracked:  45_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-BDCL_filtered_PSMs.tsv
    Untracked:  45_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  46_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-DCL_filtered_PSMs.tsv
    Untracked:  46_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-DCL_filtered_PSMs.xlsx
    Untracked:  47_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-SRF_filtered_PSMs.tsv
    Untracked:  47_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-SRF_filtered_PSMs.xlsx
    Untracked:  48_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-BDCL_filtered_PSMs.tsv
    Untracked:  48_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  49_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-DCL_filtered_PSMs.tsv
    Untracked:  49_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-DCL_filtered_PSMs.xlsx
    Untracked:  50_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-SRF_filtered_PSMs.tsv
    Untracked:  50_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-SRF_filtered_PSMs.xlsx
    Untracked:  51_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-BDCL_filtered_PSMs.tsv
    Untracked:  51_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  52_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-DCL_filtered_PSMs.tsv
    Untracked:  52_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-DCL_filtered_PSMs.xlsx
    Untracked:  53_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-SRF_filtered_PSMs.tsv
    Untracked:  53_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-SRF_filtered_PSMs.xlsx
    Untracked:  54_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-BDCL_filtered_PSMs.tsv
    Untracked:  54_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  55_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-DCL_filtered_PSMs.tsv
    Untracked:  55_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-DCL_filtered_PSMs.xlsx
    Untracked:  56_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-SRF_filtered_PSMs.tsv
    Untracked:  56_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-SRF_filtered_PSMs.xlsx
    Untracked:  57_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-BDCL_filtered_PSMs.tsv
    Untracked:  57_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  58_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-DCL_filtered_PSMs.tsv
    Untracked:  58_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-DCL_filtered_PSMs.xlsx
    Untracked:  59_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-SRF_filtered_PSMs.tsv
    Untracked:  59_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-SRF_filtered_PSMs.xlsx
    Untracked:  60_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-BDCL_filtered_PSMs.tsv
    Untracked:  60_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  61_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-DCL_filtered_PSMs.tsv
    Untracked:  61_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-DCL_filtered_PSMs.xlsx
    Untracked:  62_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-SRF_filtered_PSMs.tsv
    Untracked:  62_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-SRF_filtered_PSMs.xlsx
    Untracked:  63_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-BDCL_filtered_PSMs.tsv
    Untracked:  63_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-BDCL_filtered_PSMs.xlsx
    Untracked:  64_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-DCL_filtered_PSMs.tsv
    Untracked:  64_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-DCL_filtered_PSMs.xlsx
    Untracked:  65_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-BDCL_filtered_PSMs.tsv
    Untracked:  65_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  66_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-DCL_filtered_PSMs.tsv
    Untracked:  66_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-DCL_filtered_PSMs.xlsx
    Untracked:  67_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-SRF_filtered_PSMs.tsv
    Untracked:  67_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-SRF_filtered_PSMs.xlsx
    Untracked:  Depth_tax_12_15.pdf
    Untracked:  Hist_freq_of_peptide_id.pdf
    Untracked:  Hist_intensity_replicate_centered.pdf
    Untracked:  Hist_intensity_replicate_not_centered.pdf
    Untracked:  Hist_intensity_spec_level_centered.pdf
    Untracked:  Hist_intensity_spec_level_not_centered.pdf
    Untracked:  Hist_number of timepoints_peptide_appears.pdf
    Untracked:  P1_number_unique_peptide_per_sample.pdf
    Untracked:  P2_number_of_cumulative_unique_peptides.pdf
    Untracked:  Peptide_id.tsv
    Untracked:  Replicate_ABC50_combined.tsv
    Untracked:  Replicate_ABC50_combined.xlsx
    Untracked:  Replicate_ABC63_combined.tsv
    Untracked:  Replicate_ABC63_combined.xlsx
    Untracked:  Replicate_ABC67_combined.tsv
    Untracked:  Replicate_ABC67_combined.xlsx
    Untracked:  Replicate_BC_combined.tsv
    Untracked:  Replicate_BC_combined.xlsx
    Untracked:  Replicate_B_combined.tsv
    Untracked:  Replicate_B_combined.xlsx
    Untracked:  SRF_Genus_15_18.pdf
    Untracked:  SRF_family_12_15.pdf
    Untracked:  SRF_order_12_15.pdf
    Untracked:  Tax_strain.pdf
    Untracked:  Tax_strain_24_31.pdf
    Untracked:  Taxonomy_for_rhythmic_peptide.pdf
    Untracked:  Taxonomy_genus.pdf
    Untracked:  Taxonomy_order.pdf
    Untracked:  Taxonomy_phylum.pdf
    Untracked:  Taxonomy_species.pdf
    Untracked:  Taxonomy_strain.pdf
    Untracked:  Taxonoy_family.pdf
    Untracked:  Time_series_intesity_all.pdf
    Untracked:  data/Final_pep_seq_ORF_rankedlineage.tsv
    Untracked:  data/Time_metadata.txt
    Untracked:  data/counts_example.csv
    Untracked:  data/diel-Feb21/
    Untracked:  data/diel-Feb25/
    Untracked:  data/diel-Feb26/
    Untracked:  data/diel-Mar2/
    Untracked:  data/synthetic_control.data.txt
    Untracked:  hclust_DTW_average_wide.pdf
    Untracked:  hclust_DTW_average_wide_after_normalization.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/convert_tsv_to_dataframe.Rmd</code>) and HTML (<code>docs/convert_tsv_to_dataframe.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/fb4d76396edc94e7c43cff8cad6db42ebadbaabf/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">fb4d763</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-18
</td>
<td>
Rmd total upgrade
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/3ca3689575a1ce177980d5e27a35bbdfd6c7457e/docs/convert_tsv_to_dataframe.html" target="_blank">3ca3689</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/3b32cf64c2bdcc63a2a6109fa1b697f4b95f59b8/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">3b32cf6</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
important
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/0a00f13b81438dbaa8808aaed1a59e77a9cad43a/docs/convert_tsv_to_dataframe.html" target="_blank">0a00f13</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/52d2f3cc2eeddf54b96a8fd538594cbc2acd6452/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">52d2f3c</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-08
</td>
<td>
reversed intensity and centered by subtracting median
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/52cb18451f2c50292d084abf3126d330ef6be2bb/docs/convert_tsv_to_dataframe.html" target="_blank">52cb184</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7f26b0f0ed5a088087ff67171af9bd22c65efe5f/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">7f26b0f</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/c2f25d3613813ebed224e4e969c5753250e890fe/docs/convert_tsv_to_dataframe.html" target="_blank">c2f25d3</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/c5ec37ef6ebd29a83bdba8d242b57fccc7837df9/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">c5ec37e</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-26
</td>
<td>
total 50 samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/3db29b0bc2c1673a15f5600c103841a88b5a47bf/docs/convert_tsv_to_dataframe.html" target="_blank">3db29b0</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/fe508d15413c128cce2d577485c74c32d77d1704/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">fe508d1</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/e55579c873cb14ccd7bc7796c410ba177abc2a6c/docs/convert_tsv_to_dataframe.html" target="_blank">e55579c</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/44eb05fc0b315c8860224b5d2d2d40e0947e21dd/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">44eb05f</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/12b2c74add495b09b165c27e24022cdd0f863103/docs/convert_tsv_to_dataframe.html" target="_blank">12b2c74</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/c4cbcd229432d571f38ef352a2c584f61be1c6c1/docs/convert_tsv_to_dataframe.html" target="_blank">c4cbcd2</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7830f0053fe892c583494bdfddfc272fe7ffbd2c/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">7830f00</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/366e63ad3f1783c91278abbe68d0b0e2b734aae2/docs/convert_tsv_to_dataframe.html" target="_blank">366e63a</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/993544907145f25b0891066dca24997732b39810/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">9935449</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/f7aefb080f623efe866012fa39847e15313bfd6a/docs/convert_tsv_to_dataframe.html" target="_blank">f7aefb0</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/6b63e24d0a19f068fbc4e92d85f740a8b1632764/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">6b63e24</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/6ea68a204382c474fc5eec4d2c323c6c7810c06c/docs/convert_tsv_to_dataframe.html" target="_blank">6ea68a2</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/f5d311b6e4a50c52f72d519dbcfdb47cef69517d/analysis/convert_tsv_to_dataframe.Rmd" target="_blank">f5d311b</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
publish
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="converting-tsv-files-to-dataframe" class="section level3">
<h3>Converting tsv files to dataframe</h3>
<p>It is very important to change I to L.</p>
<pre class="r"><code># libraries
library(dplyr)
library(tidyr)
library(stringr)
library(ggplot2)

# Make the desirable format
df &lt;- read.csv(file=&#39;data/diel-Mar2/IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.tsv&#39;, sep = &#39;\t&#39;)
df &lt;- tibble::rownames_to_column(df, var = &#39;Index&#39;)
df2 &lt;- df %&gt;% select(Index,Scan.Number, Base.Peptide.Sequence, log2.18O.16O.Ratio, Target., Decoy.)
# change I to L
sum(grepl(&quot;I&quot;, df2$Base.Peptide.Sequence))</code></pre>
<pre><code>[1] 892</code></pre>
<pre class="r"><code>df2$Base.Peptide.Sequence &lt;- gsub(&quot;I&quot;,&quot;L&quot;,df2$Base.Peptide.Sequence) 
sum(grepl(&quot;I&quot;, df2$Base.Peptide.Sequence))</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>df3 &lt;- df2 %&gt;% separate(Scan.Number, into = c(&#39;dir&#39;,&quot;file_name&quot;), sep = &#39;IPTL-&#39;)
colnames(df3)</code></pre>
<pre><code>[1] &quot;Index&quot;                 &quot;dir&quot;                   &quot;file_name&quot;            
[4] &quot;Base.Peptide.Sequence&quot; &quot;log2.18O.16O.Ratio&quot;    &quot;Target.&quot;              
[7] &quot;Decoy.&quot;               </code></pre>
<pre class="r"><code>df4 &lt;- df3 %&gt;% select(-dir)
colnames(df4)</code></pre>
<pre><code>[1] &quot;Index&quot;                 &quot;file_name&quot;             &quot;Base.Peptide.Sequence&quot;
[4] &quot;log2.18O.16O.Ratio&quot;    &quot;Target.&quot;               &quot;Decoy.&quot;               </code></pre>
<pre class="r"><code>df5 &lt;- df4 %&gt;% separate(file_name, sep=&#39;;&#39;, into= c(&#39;IPT&#39;,&#39;SpectrumID&#39;,&#39;ScanID&#39;))
head(df5)</code></pre>
<pre><code>  Index                                      IPT         SpectrumID
1     1 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 49728
2     2 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 42413
3     3 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 43799
4     4 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 41495
5     5 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 42258
6     6 CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL.raw  SpectrumID: 49943
         ScanID Base.Peptide.Sequence log2.18O.16O.Ratio Target. Decoy.
1  scans: 56549  LLSENVDALLGAASSGVSLK        -0.60046965    True  False
2  scans: 48736    SVFVNQASSEGLGQALAK        -0.09393621    True  False
3  scans: 50215    EELASVASLSAADTELGR        -1.17728746    True  False
4  scans: 47756    GQADLGLLGSTAGTQLNR        -0.82806202    True  False
5  scans: 48571    SVFVNQASSEGLGQALAK        -1.05946785    True  False
6  scans: 56779 AMLQDLAVLTGGTVLTDELGR        -0.70018147    True  False</code></pre>
<pre class="r"><code># remove unwanted substrings
df5$IPT &lt;- str_remove(df5$IPT, &#39;.raw&#39;)
df5$SpectrumID &lt;- str_remove(df5$SpectrumID, &#39; SpectrumID: &#39;)
df5$ScanID &lt;- str_remove(df5$ScanID, &#39; scans: &#39;) 
head(df5$ScanID) # watch out for the space in front</code></pre>
<pre><code>[1] &quot;56549&quot; &quot;48736&quot; &quot;50215&quot; &quot;47756&quot; &quot;48571&quot; &quot;56779&quot;</code></pre>
<pre class="r"><code>df6 &lt;- df5 %&gt;% separate(IPT,into=c(&#39;Cruise&#39;,&#39;Station&#39;,&#39;Isotope&#39;,&#39;16O&#39;,&#39;Replicate&#39;,&#39;Time&#39;,&#39;Depth&#39;),sep=&#39;-&#39;) %&gt;% select(-`16O`)
colnames(df6)</code></pre>
<pre><code> [1] &quot;Index&quot;                 &quot;Cruise&quot;                &quot;Station&quot;              
 [4] &quot;Isotope&quot;               &quot;Replicate&quot;             &quot;Time&quot;                 
 [7] &quot;Depth&quot;                 &quot;SpectrumID&quot;            &quot;ScanID&quot;               
[10] &quot;Base.Peptide.Sequence&quot; &quot;log2.18O.16O.Ratio&quot;    &quot;Target.&quot;              
[13] &quot;Decoy.&quot;               </code></pre>
<pre class="r"><code>df7 &lt;- df6 %&gt;% unite(c(&#39;Depth&#39;,&#39;Time&#39;,&#39;Replicate&#39;,&#39;ScanID&#39;),col=Spec_id, remove=F) 
df7 &lt;- df7 %&gt;% unite(c(&#39;Depth&#39;,&#39;Time&#39;,&#39;Replicate&#39;),col=SampleID, remove=F) 
head(df7)</code></pre>
<pre><code>  Index    Cruise Station Isotope         Spec_id  SampleID Replicate Time
1     1 CyanoSUMI   MI41M  18OMix BDCL_T1_B_56549 BDCL_T1_B         B   T1
2     2 CyanoSUMI   MI41M  18OMix BDCL_T1_B_48736 BDCL_T1_B         B   T1
3     3 CyanoSUMI   MI41M  18OMix BDCL_T1_B_50215 BDCL_T1_B         B   T1
4     4 CyanoSUMI   MI41M  18OMix BDCL_T1_B_47756 BDCL_T1_B         B   T1
5     5 CyanoSUMI   MI41M  18OMix BDCL_T1_B_48571 BDCL_T1_B         B   T1
6     6 CyanoSUMI   MI41M  18OMix BDCL_T1_B_56779 BDCL_T1_B         B   T1
  Depth SpectrumID ScanID Base.Peptide.Sequence log2.18O.16O.Ratio Target.
1  BDCL      49728  56549  LLSENVDALLGAASSGVSLK        -0.60046965    True
2  BDCL      42413  48736    SVFVNQASSEGLGQALAK        -0.09393621    True
3  BDCL      43799  50215    EELASVASLSAADTELGR        -1.17728746    True
4  BDCL      41495  47756    GQADLGLLGSTAGTQLNR        -0.82806202    True
5  BDCL      42258  48571    SVFVNQASSEGLGQALAK        -1.05946785    True
6  BDCL      49943  56779 AMLQDLAVLTGGTVLTDELGR        -0.70018147    True
  Decoy.
1  False
2  False
3  False
4  False
5  False
6  False</code></pre>
<pre class="r"><code># important to reverse the ratio
df8 &lt;- df7 %&gt;% mutate(log2.16O.18O.Ratio = -log2.18O.16O.Ratio)

# Arrange column order
df9 &lt;- df8 %&gt;% select(Spec_id, Base.Peptide.Sequence, log2.16O.18O.Ratio,SampleID,Index,Cruise,Station,Isotope, Replicate, Time, Depth, SpectrumID, ScanID, Target., Decoy.)
head(df9)   </code></pre>
<pre><code>          Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio  SampleID Index
1 BDCL_T1_B_56549  LLSENVDALLGAASSGVSLK         0.60046965 BDCL_T1_B     1
2 BDCL_T1_B_48736    SVFVNQASSEGLGQALAK         0.09393621 BDCL_T1_B     2
3 BDCL_T1_B_50215    EELASVASLSAADTELGR         1.17728746 BDCL_T1_B     3
4 BDCL_T1_B_47756    GQADLGLLGSTAGTQLNR         0.82806202 BDCL_T1_B     4
5 BDCL_T1_B_48571    SVFVNQASSEGLGQALAK         1.05946785 BDCL_T1_B     5
6 BDCL_T1_B_56779 AMLQDLAVLTGGTVLTDELGR         0.70018147 BDCL_T1_B     6
     Cruise Station Isotope Replicate Time Depth SpectrumID ScanID Target.
1 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      49728  56549    True
2 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      42413  48736    True
3 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      43799  50215    True
4 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      41495  47756    True
5 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      42258  48571    True
6 CyanoSUMI   MI41M  18OMix         B   T1  BDCL      49943  56779    True
  Decoy.
1  False
2  False
3  False
4  False
5  False
6  False</code></pre>
<pre class="r"><code># screen and remove Decoy = True
df10 &lt;- df9 %&gt;% filter(Decoy.==&#39;False&#39;)
# confirm
df10[df10$Decoy.==&#39;True&#39;,]</code></pre>
<pre><code> [1] Spec_id               Base.Peptide.Sequence log2.16O.18O.Ratio   
 [4] SampleID              Index                 Cruise               
 [7] Station               Isotope               Replicate            
[10] Time                  Depth                 SpectrumID           
[13] ScanID                Target.               Decoy.               
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code># subtract median the intensity after filtering =&gt; output change to cent.log2.16O.18O.Ratio (centered)
median(df10$log2.16O.18O.Ratio)</code></pre>
<pre><code>[1] 0.6346337</code></pre>
<pre class="r"><code>df11 &lt;- df10 %&gt;% mutate(cent.log2.16O.18O.Ratio = log2.16O.18O.Ratio - median(df10$log2.16O.18O.Ratio)) %&gt;%
  select(Spec_id, Base.Peptide.Sequence, log2.16O.18O.Ratio,cent.log2.16O.18O.Ratio, SampleID,Index,Cruise,Station,Isotope, Replicate, Time, Depth, SpectrumID, ScanID, Target., Decoy.)
# final form
head(df11)</code></pre>
<pre><code>          Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio
1 BDCL_T1_B_56549  LLSENVDALLGAASSGVSLK         0.60046965
2 BDCL_T1_B_48736    SVFVNQASSEGLGQALAK         0.09393621
3 BDCL_T1_B_50215    EELASVASLSAADTELGR         1.17728746
4 BDCL_T1_B_47756    GQADLGLLGSTAGTQLNR         0.82806202
5 BDCL_T1_B_48571    SVFVNQASSEGLGQALAK         1.05946785
6 BDCL_T1_B_56779 AMLQDLAVLTGGTVLTDELGR         0.70018147
  cent.log2.16O.18O.Ratio  SampleID Index    Cruise Station Isotope Replicate
1             -0.03416409 BDCL_T1_B     1 CyanoSUMI   MI41M  18OMix         B
2             -0.54069753 BDCL_T1_B     2 CyanoSUMI   MI41M  18OMix         B
3              0.54265372 BDCL_T1_B     3 CyanoSUMI   MI41M  18OMix         B
4              0.19342828 BDCL_T1_B     4 CyanoSUMI   MI41M  18OMix         B
5              0.42483411 BDCL_T1_B     5 CyanoSUMI   MI41M  18OMix         B
6              0.06554773 BDCL_T1_B     6 CyanoSUMI   MI41M  18OMix         B
  Time Depth SpectrumID ScanID Target. Decoy.
1   T1  BDCL      49728  56549    True  False
2   T1  BDCL      42413  48736    True  False
3   T1  BDCL      43799  50215    True  False
4   T1  BDCL      41495  47756    True  False
5   T1  BDCL      42258  48571    True  False
6   T1  BDCL      49943  56779    True  False</code></pre>
</div>
<div id="doing-it-for-all-tsv-files" class="section level3">
<h3>Doing it for all tsv files</h3>
<pre class="r"><code>library(openxlsx)

# define function
convert_tsv &lt;- function(file_number, filename){
  # Make the desirable format
  df &lt;- read.csv(file=paste0(directory_path,&quot;/&quot; ,filename), sep = &#39;\t&#39;)
  df &lt;- tibble::rownames_to_column(df, var = &#39;Index&#39;)
  df2 &lt;- df %&gt;% select(Index,Scan.Number, Base.Peptide.Sequence, log2.18O.16O.Ratio, Target., Decoy.)
  # change I to L
  df2$Base.Peptide.Sequence &lt;- gsub(&quot;I&quot;,&quot;L&quot;,df2$Base.Peptide.Sequence) 
  df3 &lt;- df2 %&gt;% separate(Scan.Number, into = c(&#39;dir&#39;,&quot;file_name&quot;), sep = &#39;IPTL-&#39;)
  colnames(df3)
  df4 &lt;- df3 %&gt;% select(-dir)
  colnames(df4)
  df5 &lt;- df4 %&gt;% separate(file_name, sep=&#39;;&#39;, into= c(&#39;IPT&#39;,&#39;SpectrumID&#39;,&#39;ScanID&#39;))
  head(df5)
  # remove unwanted substrings
  df5$IPT &lt;- str_remove(df5$IPT, &#39;.raw&#39;)
  df5$SpectrumID &lt;- str_remove(df5$SpectrumID, &#39; SpectrumID: &#39;)
  df5$ScanID &lt;- str_remove(df5$ScanID, &#39; scans: &#39;) 
  head(df5$ScanID) # watch out for the space in front
  df6 &lt;- df5 %&gt;% separate(IPT,into=c(&#39;Cruise&#39;,&#39;Station&#39;,&#39;Isotope&#39;,&#39;16O&#39;,&#39;Replicate&#39;,&#39;Time&#39;,&#39;Depth&#39;),sep=&#39;-&#39;) %&gt;% select(-`16O`)
  colnames(df6)
  df7 &lt;- df6 %&gt;% unite(c(&#39;Depth&#39;,&#39;Time&#39;,&#39;Replicate&#39;,&#39;ScanID&#39;),col=Spec_id, remove=F) 
  df7 &lt;- df7 %&gt;% unite(c(&#39;Depth&#39;,&#39;Time&#39;,&#39;Replicate&#39;),col=SampleID, remove=F) 
  head(df7)
  # important to reverse the ratio
  df8 &lt;- df7 %&gt;% mutate(log2.16O.18O.Ratio = -log2.18O.16O.Ratio)
  
  # Arrange column order
  df9 &lt;- df8 %&gt;% select(Spec_id, Base.Peptide.Sequence, log2.16O.18O.Ratio,SampleID,Index,Cruise,Station,Isotope, Replicate, Time, Depth, SpectrumID, ScanID, Target., Decoy.)
  head(df9)   
  # screen and remove Decoy = True
  df10 &lt;- df9 %&gt;% filter(Decoy.==&#39;False&#39;)
  # confirm
  df10[df10$Decoy.==&#39;True&#39;,]
  
  # subtract median the intensity after filtering =&gt; output change to cent.log2.16O.18O.Ratio (centered)
  median(df10$log2.16O.18O.Ratio)
  df11 &lt;- df10 %&gt;% mutate(cent.log2.16O.18O.Ratio = log2.16O.18O.Ratio - median(df10$log2.16O.18O.Ratio)) %&gt;%
    select(Spec_id, Base.Peptide.Sequence, log2.16O.18O.Ratio,cent.log2.16O.18O.Ratio, SampleID,Index,Cruise,Station,Isotope, Replicate, Time, Depth, SpectrumID, ScanID, Target., Decoy.)
  # final form
  head(df11)
  write.table(df11,file=paste0(file_number,&#39;_&#39;,filename), sep=&#39;\t&#39;,row.names = F)
  openxlsx::write.xlsx(df11,paste0(file_number,&#39;_&#39;,str_remove(filename,&#39;.tsv&#39;),&#39;.xlsx&#39;),rowNames=F)
}

# list all files in certain folder
directory_path = &quot;data/diel-Mar2&quot;
list_f &lt;- list.files(path=&quot;data/diel-Mar2&quot;)
length(list_f)</code></pre>
<pre><code>[1] 67</code></pre>
<pre class="r"><code># test example
# convert_tsv(&#39;01&#39;,&#39;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.tsv&#39;)

# make file name starting from 01, 02, 
vec_00 &lt;- sprintf(&quot;%02d&quot;, c(1:length(list_f)))
# loop through files and convert
for (i in c(1:length(list_f))){
  print(i)
  print(list_f[i])
  convert_tsv(vec_00[i], list_f[i])
}</code></pre>
<pre><code>[1] 1
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-BDCL_filtered_PSMs.tsv&quot;
[1] 2
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-DCL_filtered_PSMs.tsv&quot;
[1] 3
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-SRF_filtered_PSMs.tsv&quot;
[1] 4
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-BDCL_filtered_PSMs.tsv&quot;
[1] 5
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-DCL_filtered_PSMs.tsv&quot;
[1] 6
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-SRF_filtered_PSMs.tsv&quot;
[1] 7
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-BDCL_filtered_PSMs.tsv&quot;
[1] 8
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-DCL_filtered_PSMs.tsv&quot;
[1] 9
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-SRF_filtered_PSMs.tsv&quot;
[1] 10
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-BDCL_filtered_PSMs.tsv&quot;
[1] 11
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-DCL_filtered_PSMs.tsv&quot;
[1] 12
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-SRF_filtered_PSMs.tsv&quot;
[1] 13
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-BDCL_filtered_PSMs.tsv&quot;
[1] 14
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-DCL_filtered_PSMs.tsv&quot;
[1] 15
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-SRF_filtered_PSMs.tsv&quot;
[1] 16
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-BDCL_filtered_PSMs.tsv&quot;
[1] 17
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-DCL_filtered_PSMs.tsv&quot;
[1] 18
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-SRF_filtered_PSMs.tsv&quot;
[1] 19
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-DCL_filtered_PSMs.tsv&quot;
[1] 20
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-SRF_filtered_PSMs.tsv&quot;
[1] 21
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-BDCL_filtered_PSMs.tsv&quot;
[1] 22
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-DCL_filtered_PSMs.tsv&quot;
[1] 23
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-SRF_filtered_PSMs.tsv&quot;
[1] 24
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.tsv&quot;
[1] 25
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-DCL_filtered_PSMs.tsv&quot;
[1] 26
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-SRF_filtered_PSMs.tsv&quot;
[1] 27
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-BDCL_filtered_PSMs.tsv&quot;
[1] 28
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-DCL_filtered_PSMs.tsv&quot;
[1] 29
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-SRF_filtered_PSMs.tsv&quot;
[1] 30
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-BDCL_filtered_PSMs.tsv&quot;
[1] 31
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-DCL_filtered_PSMs.tsv&quot;
[1] 32
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-SRF_filtered_PSMs.tsv&quot;
[1] 33
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-BDCL_filtered_PSMs.tsv&quot;
[1] 34
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-DCL_filtered_PSMs.tsv&quot;
[1] 35
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-SRF_filtered_PSMs.tsv&quot;
[1] 36
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-BDCL_filtered_PSMs.tsv&quot;
[1] 37
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-DCL_filtered_PSMs.tsv&quot;
[1] 38
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-SRF_filtered_PSMs.tsv&quot;
[1] 39
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-BDCL_filtered_PSMs.tsv&quot;
[1] 40
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-DCL_filtered_PSMs.tsv&quot;
[1] 41
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-SRF_filtered_PSMs.tsv&quot;
[1] 42
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-BDCL_filtered_PSMs.tsv&quot;
[1] 43
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-DCL_filtered_PSMs.tsv&quot;
[1] 44
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-SRF_filtered_PSMs.tsv&quot;
[1] 45
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-BDCL_filtered_PSMs.tsv&quot;
[1] 46
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-DCL_filtered_PSMs.tsv&quot;
[1] 47
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-SRF_filtered_PSMs.tsv&quot;
[1] 48
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-BDCL_filtered_PSMs.tsv&quot;
[1] 49
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-DCL_filtered_PSMs.tsv&quot;
[1] 50
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-SRF_filtered_PSMs.tsv&quot;
[1] 51
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-BDCL_filtered_PSMs.tsv&quot;
[1] 52
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-DCL_filtered_PSMs.tsv&quot;
[1] 53
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-SRF_filtered_PSMs.tsv&quot;
[1] 54
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-BDCL_filtered_PSMs.tsv&quot;
[1] 55
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-DCL_filtered_PSMs.tsv&quot;
[1] 56
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-SRF_filtered_PSMs.tsv&quot;
[1] 57
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-BDCL_filtered_PSMs.tsv&quot;
[1] 58
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-DCL_filtered_PSMs.tsv&quot;
[1] 59
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-SRF_filtered_PSMs.tsv&quot;
[1] 60
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-BDCL_filtered_PSMs.tsv&quot;
[1] 61
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-DCL_filtered_PSMs.tsv&quot;
[1] 62
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-SRF_filtered_PSMs.tsv&quot;
[1] 63
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-BDCL_filtered_PSMs.tsv&quot;
[1] 64
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-DCL_filtered_PSMs.tsv&quot;
[1] 65
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-BDCL_filtered_PSMs.tsv&quot;
[1] 66
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-DCL_filtered_PSMs.tsv&quot;
[1] 67
[1] &quot;IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-SRF_filtered_PSMs.tsv&quot;</code></pre>
</div>
<div id="aggregate-all-converted-tsv-files-make-peptide_id-for-all-unique-pep-sequences" class="section level3">
<h3>Aggregate all converted tsv files &amp; make Peptide_id for all unique pep sequences</h3>
<pre class="r"><code>#
tsv_files  &lt;- list.files(pattern = &#39;^[0-9].*_PSMs\\.tsv&#39;)
tables &lt;- lapply(tsv_files, read.table, header = TRUE)
combined_tsv &lt;- do.call(rbind , tables)
head(combined_tsv)</code></pre>
<pre><code>          Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio
1 BDCL_T1_A_39859    NTVGGGLPVMTGPGFVDK          0.5928105
2 BDCL_T1_A_45254    SVFVNQASSEGLGQALAK          1.0214321
3 BDCL_T1_A_39843    NTVGGGLPVMTGPGFVDK          0.4909865
4 BDCL_T1_A_34716       GFGFVEMDSQEGAQK          0.6611921
5 BDCL_T1_A_50767      LLDQGQAGDNVGLLLR          0.8198039
6 BDCL_T1_A_44469    GQADLGLLGSTAGTQLNR          0.8216291
  cent.log2.16O.18O.Ratio  SampleID Index    Cruise Station Isotope Replicate
1             -0.18097898 BDCL_T1_A     1 CyanoSUMI   MI41M  18OMix         A
2              0.24764266 BDCL_T1_A     2 CyanoSUMI   MI41M  18OMix         A
3             -0.28280292 BDCL_T1_A     3 CyanoSUMI   MI41M  18OMix         A
4             -0.11259732 BDCL_T1_A     4 CyanoSUMI   MI41M  18OMix         A
5              0.04601442 BDCL_T1_A     5 CyanoSUMI   MI41M  18OMix         A
6              0.04783963 BDCL_T1_A     6 CyanoSUMI   MI41M  18OMix         A
  Time Depth SpectrumID ScanID Target. Decoy.
1   T1  BDCL      34516  39859    True  False
2   T1  BDCL      39569  45254    True  False
3   T1  BDCL      34501  39843    True  False
4   T1  BDCL      29699  34716    True  False
5   T1  BDCL      44732  50767    True  False
6   T1  BDCL      38834  44469    True  False</code></pre>
<pre class="r"><code>dim(combined_tsv)</code></pre>
<pre><code>[1] 96551    16</code></pre>
<pre class="r"><code># give unique peptide id&#39;s for all peptides
# give peptide name by frequency (peptide with greatest frequency gets Pep_1)
df_p &lt;- transform(combined_tsv, freq= ave(seq(nrow(combined_tsv)), Base.Peptide.Sequence, FUN=length))
df_p[order(-df_p$freq), ] %&gt;% head()</code></pre>
<pre><code>            Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio
25  BDCL_T1_A_31195       AELLDAMAAGSGLSK          0.3969800
49  BDCL_T1_A_31179       AELLDAMAAGSGLSK          0.5078004
69  BDCL_T1_A_31003       AELLDAMAAGSGLSK          0.3509932
72  BDCL_T1_A_31022       AELLDAMAAGSGLSK          0.4062575
142 BDCL_T1_A_31372       AELLDAMAAGSGLSK          0.4000644
180 BDCL_T1_A_31355       AELLDAMAAGSGLSK          0.3204865
    cent.log2.16O.18O.Ratio  SampleID Index    Cruise Station Isotope Replicate
25               -0.3768094 BDCL_T1_A    25 CyanoSUMI   MI41M  18OMix         A
49               -0.2659891 BDCL_T1_A    49 CyanoSUMI   MI41M  18OMix         A
69               -0.4227962 BDCL_T1_A    69 CyanoSUMI   MI41M  18OMix         A
72               -0.3675320 BDCL_T1_A    72 CyanoSUMI   MI41M  18OMix         A
142              -0.3737250 BDCL_T1_A   142 CyanoSUMI   MI41M  18OMix         A
180              -0.4533029 BDCL_T1_A   180 CyanoSUMI   MI41M  18OMix         A
    Time Depth SpectrumID ScanID Target. Decoy. freq
25    T1  BDCL      26401  31195    True  False  825
49    T1  BDCL      26386  31179    True  False  825
69    T1  BDCL      26221  31003    True  False  825
72    T1  BDCL      26239  31022    True  False  825
142   T1  BDCL      26567  31372    True  False  825
180   T1  BDCL      26551  31355    True  False  825</code></pre>
<pre class="r"><code>df_pepid &lt;- df_p %&gt;% select(Base.Peptide.Sequence,freq) %&gt;% arrange(desc(freq)) %&gt;% unique() %&gt;% arrange()

# Distribution of frequency of each Peptide_id in all samples
theme_set(theme_bw())
his1 &lt;- ggplot(df_pepid, aes(x=freq)) +
  geom_histogram(binwidth = 10) +
  scale_fill_brewer(palette=&#39;Set2&#39;) +
  ylab(&quot;Frequency \n&quot;) +
  xlab(&quot;\n Frequency of each unique peptide sequence&quot;) +
  ggtitle(&quot;Distribution of frequency of each Peptide_id in all samples \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))
  # theme(panel.grid.major = element_blank())
  # theme(panel.grid.minor = element_blank())
his1</code></pre>
<p><img src="figure/convert_tsv_to_dataframe.Rmd/unnamed-chunk-3-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>df_pepid2 &lt;- tibble::rownames_to_column(df_pepid, var=&#39;rowid&#39;) 
df_pepid3 &lt;- df_pepid2 %&gt;% mutate(Peptide_id = paste0(&quot;Pep_&quot;,rowid)) %&gt;% select(-rowid)
head(df_pepid3)</code></pre>
<pre><code>  Base.Peptide.Sequence freq Peptide_id
1       AELLDAMAAGSGLSK  825      Pep_1
2         TYQGPATGVLVER  793      Pep_2
3      SVVTLSGSMGEVLLGR  784      Pep_3
4       GFGFVEMDSQEGAQK  774      Pep_4
5       SVVTTVVTAADAAGR  773      Pep_5
6            FLFGMEGVNR  698      Pep_6</code></pre>
<pre class="r"><code>tail(df_pepid3)</code></pre>
<pre><code>     Base.Peptide.Sequence freq Peptide_id
3321              TKDGALAR    1   Pep_3321
3322           AAATGSLGDGK    1   Pep_3322
3323              LAASSGAH    1   Pep_3323
3324           EVTGGGGPGGK    1   Pep_3324
3325            ASSEGKGGAK    1   Pep_3325
3326              KSVDVNPK    1   Pep_3326</code></pre>
<pre class="r"><code># check if sum of freq is same as all number of spectral reads
sum(df_pepid3$freq) == dim(combined_tsv)[1]</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># write Peptide_id list
write.table(df_pepid3,&quot;Peptide_id.tsv&quot;, sep=&#39;\t&#39;,row.names = F)

combined_tsv_pepid &lt;- combined_tsv %&gt;% left_join(df_pepid3, by=c(&#39;Base.Peptide.Sequence&#39;=&#39;Base.Peptide.Sequence&#39;)) %&gt;% select(-freq)

colnames(combined_tsv_pepid)</code></pre>
<pre><code> [1] &quot;Spec_id&quot;                 &quot;Base.Peptide.Sequence&quot;  
 [3] &quot;log2.16O.18O.Ratio&quot;      &quot;cent.log2.16O.18O.Ratio&quot;
 [5] &quot;SampleID&quot;                &quot;Index&quot;                  
 [7] &quot;Cruise&quot;                  &quot;Station&quot;                
 [9] &quot;Isotope&quot;                 &quot;Replicate&quot;              
[11] &quot;Time&quot;                    &quot;Depth&quot;                  
[13] &quot;SpectrumID&quot;              &quot;ScanID&quot;                 
[15] &quot;Target.&quot;                 &quot;Decoy.&quot;                 
[17] &quot;Peptide_id&quot;             </code></pre>
<pre class="r"><code>head(combined_tsv_pepid)</code></pre>
<pre><code>          Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio
1 BDCL_T1_A_39859    NTVGGGLPVMTGPGFVDK          0.5928105
2 BDCL_T1_A_45254    SVFVNQASSEGLGQALAK          1.0214321
3 BDCL_T1_A_39843    NTVGGGLPVMTGPGFVDK          0.4909865
4 BDCL_T1_A_34716       GFGFVEMDSQEGAQK          0.6611921
5 BDCL_T1_A_50767      LLDQGQAGDNVGLLLR          0.8198039
6 BDCL_T1_A_44469    GQADLGLLGSTAGTQLNR          0.8216291
  cent.log2.16O.18O.Ratio  SampleID Index    Cruise Station Isotope Replicate
1             -0.18097898 BDCL_T1_A     1 CyanoSUMI   MI41M  18OMix         A
2              0.24764266 BDCL_T1_A     2 CyanoSUMI   MI41M  18OMix         A
3             -0.28280292 BDCL_T1_A     3 CyanoSUMI   MI41M  18OMix         A
4             -0.11259732 BDCL_T1_A     4 CyanoSUMI   MI41M  18OMix         A
5              0.04601442 BDCL_T1_A     5 CyanoSUMI   MI41M  18OMix         A
6              0.04783963 BDCL_T1_A     6 CyanoSUMI   MI41M  18OMix         A
  Time Depth SpectrumID ScanID Target. Decoy. Peptide_id
1   T1  BDCL      34516  39859    True  False     Pep_32
2   T1  BDCL      39569  45254    True  False    Pep_231
3   T1  BDCL      34501  39843    True  False     Pep_32
4   T1  BDCL      29699  34716    True  False      Pep_4
5   T1  BDCL      44732  50767    True  False     Pep_67
6   T1  BDCL      38834  44469    True  False    Pep_130</code></pre>
<pre class="r"><code>tail(combined_tsv_pepid)</code></pre>
<pre><code>             Spec_id Base.Peptide.Sequence log2.16O.18O.Ratio
96546 SRF_T8_C_36385          TVFAPTDEAFAK          0.6994099
96547 SRF_T8_C_39109         TATLGTAELASLR          0.3236297
96548 SRF_T8_C_33010         NAALGSGELAGLR          0.1417181
96549 SRF_T8_C_43445       EFVDGAWAPSAPTVK          0.3346053
96550 SRF_T8_C_52141          AVLALLNLFDDR          0.2781826
96551 SRF_T8_C_47618         AMLEDLALLTGGK          1.5152827
      cent.log2.16O.18O.Ratio SampleID Index    Cruise Station Isotope
96546             0.414845543 SRF_T8_C   201 CyanoSUMI   MI41M  18OMix
96547             0.039065332 SRF_T8_C   202 CyanoSUMI   MI41M  18OMix
96548            -0.142846328 SRF_T8_C   203 CyanoSUMI   MI41M  18OMix
96549             0.050040917 SRF_T8_C   204 CyanoSUMI   MI41M  18OMix
96550            -0.006381815 SRF_T8_C   205 CyanoSUMI   MI41M  18OMix
96551             1.230718366 SRF_T8_C   206 CyanoSUMI   MI41M  18OMix
      Replicate Time Depth SpectrumID ScanID Target. Decoy. Peptide_id
96546         C   T8   SRF      31045  36385    True  False     Pep_25
96547         C   T8   SRF      33599  39109    True  False    Pep_119
96548         C   T8   SRF      27881  33010    True  False     Pep_21
96549         C   T8   SRF      37664  43445    True  False     Pep_10
96550         C   T8   SRF      45817  52141    True  False    Pep_188
96551         C   T8   SRF      41576  47618    True  False     Pep_94</code></pre>
<pre class="r"><code># check whether there is no NA value in the dataframe
any(is.na.data.frame(combined_tsv_pepid))</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>write.table(combined_tsv_pepid,&quot;Replicate_ABC67_combined.tsv&quot;, sep=&#39;\t&#39;,row.names = F)
openxlsx::write.xlsx(combined_tsv_pepid,&#39;Replicate_ABC67_combined.xlsx&#39;,rowNames=F)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] openxlsx_4.2.3  ggplot2_3.3.3   stringr_1.4.0   tidyr_1.1.2    
[5] dplyr_1.0.4     workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] zip_2.1.1         Rcpp_1.0.5        highr_0.8         pillar_1.5.1     
 [5] compiler_4.0.3    bslib_0.2.4       later_1.1.0.1     jquerylib_0.1.3  
 [9] git2r_0.28.0      tools_4.0.3       digest_0.6.27     gtable_0.3.0     
[13] jsonlite_1.7.2    evaluate_0.14     lifecycle_1.0.0   tibble_3.0.4     
[17] pkgconfig_2.0.3   rlang_0.4.10      DBI_1.1.1         yaml_2.2.1       
[21] xfun_0.20         withr_2.4.1       knitr_1.31        generics_0.1.0   
[25] fs_1.5.0          vctrs_0.3.6       sass_0.3.1        grid_4.0.3       
[29] rprojroot_2.0.2   tidyselect_1.1.0  glue_1.4.2        R6_2.5.0         
[33] fansi_0.4.2       rmarkdown_2.7     farver_2.1.0      purrr_0.3.4      
[37] magrittr_2.0.1    whisker_0.4       scales_1.1.1      promises_1.1.1   
[41] ellipsis_0.3.1    htmltools_0.5.1.1 assertthat_0.2.1  colorspace_2.0-0 
[45] httpuv_1.5.4      labeling_0.4.2    utf8_1.1.4        stringi_1.5.3    
[49] munsell_0.5.0     crayon_1.4.1     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
