<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="KiseokUchicago" />

<meta name="date" content="2021-02-22" />

<title>peptide_level_visualization</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BH_Diel</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/KiseokUchicago/BH_Diel">
    <span class="fas fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">peptide_level_visualization</h1>
<h4 class="author">KiseokUchicago</h4>
<h4 class="date">2021-02-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>BH_Diel/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210222code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210222)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210222code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210222)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoBHDieltree2ef4a7c7cd3bf243c13749e197280bb43c8b8256targetblank2ef4a7ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/KiseokUchicago/BH_Diel/tree/2ef4a7c7cd3bf243c13749e197280bb43c8b8256" target="_blank">2ef4a7c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomKiseokUchicagoBHDieltree2ef4a7c7cd3bf243c13749e197280bb43c8b8256targetblank2ef4a7ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/KiseokUchicago/BH_Diel/tree/2ef4a7c7cd3bf243c13749e197280bb43c8b8256" target="_blank">2ef4a7c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  01_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-BDCL_filtered_PSMs.tsv
    Untracked:  01_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  02_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-DCL_filtered_PSMs.tsv
    Untracked:  02_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-DCL_filtered_PSMs.xlsx
    Untracked:  03_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-SRF_filtered_PSMs.tsv
    Untracked:  03_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T1-SRF_filtered_PSMs.xlsx
    Untracked:  04_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-BDCL_filtered_PSMs.tsv
    Untracked:  04_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-BDCL_filtered_PSMs.xlsx
    Untracked:  05_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-DCL_filtered_PSMs.tsv
    Untracked:  05_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-DCL_filtered_PSMs.xlsx
    Untracked:  06_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-SRF_filtered_PSMs.tsv
    Untracked:  06_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T2-SRF_filtered_PSMs.xlsx
    Untracked:  07_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-BDCL_filtered_PSMs.tsv
    Untracked:  07_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  08_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-DCL_filtered_PSMs.tsv
    Untracked:  08_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-DCL_filtered_PSMs.xlsx
    Untracked:  09_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-SRF_filtered_PSMs.tsv
    Untracked:  09_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T3-SRF_filtered_PSMs.xlsx
    Untracked:  10_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-BDCL_filtered_PSMs.tsv
    Untracked:  10_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  11_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-DCL_filtered_PSMs.tsv
    Untracked:  11_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-DCL_filtered_PSMs.xlsx
    Untracked:  12_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-SRF_filtered_PSMs.tsv
    Untracked:  12_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T4-SRF_filtered_PSMs.xlsx
    Untracked:  13_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-BDCL_filtered_PSMs.tsv
    Untracked:  13_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  14_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-DCL_filtered_PSMs.tsv
    Untracked:  14_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-DCL_filtered_PSMs.xlsx
    Untracked:  15_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-SRF_filtered_PSMs.tsv
    Untracked:  15_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T5-SRF_filtered_PSMs.xlsx
    Untracked:  16_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-BDCL_filtered_PSMs.tsv
    Untracked:  16_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  17_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-DCL_filtered_PSMs.tsv
    Untracked:  17_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-DCL_filtered_PSMs.xlsx
    Untracked:  18_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-SRF_filtered_PSMs.tsv
    Untracked:  18_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T6-SRF_filtered_PSMs.xlsx
    Untracked:  19_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-DCL_filtered_PSMs.tsv
    Untracked:  19_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-DCL_filtered_PSMs.xlsx
    Untracked:  20_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-SRF_filtered_PSMs.tsv
    Untracked:  20_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T7-SRF_filtered_PSMs.xlsx
    Untracked:  21_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-BDCL_filtered_PSMs.tsv
    Untracked:  21_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  22_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-DCL_filtered_PSMs.tsv
    Untracked:  22_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-DCL_filtered_PSMs.xlsx
    Untracked:  23_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-SRF_filtered_PSMs.tsv
    Untracked:  23_IPTL-CyanoSUMI-MI41M-18OMix-16O-A-T8-SRF_filtered_PSMs.xlsx
    Untracked:  24_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.tsv
    Untracked:  24_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  25_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-DCL_filtered_PSMs.tsv
    Untracked:  25_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-DCL_filtered_PSMs.xlsx
    Untracked:  26_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-SRF_filtered_PSMs.tsv
    Untracked:  26_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T1-SRF_filtered_PSMs.xlsx
    Untracked:  27_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-BDCL_filtered_PSMs.tsv
    Untracked:  27_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-BDCL_filtered_PSMs.xlsx
    Untracked:  28_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-DCL_filtered_PSMs.tsv
    Untracked:  28_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-DCL_filtered_PSMs.xlsx
    Untracked:  29_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-SRF_filtered_PSMs.tsv
    Untracked:  29_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T2-SRF_filtered_PSMs.xlsx
    Untracked:  30_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-BDCL_filtered_PSMs.tsv
    Untracked:  30_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  31_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-DCL_filtered_PSMs.tsv
    Untracked:  31_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-DCL_filtered_PSMs.xlsx
    Untracked:  32_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-SRF_filtered_PSMs.tsv
    Untracked:  32_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T3-SRF_filtered_PSMs.xlsx
    Untracked:  33_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-BDCL_filtered_PSMs.tsv
    Untracked:  33_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  34_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-DCL_filtered_PSMs.tsv
    Untracked:  34_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-DCL_filtered_PSMs.xlsx
    Untracked:  35_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-SRF_filtered_PSMs.tsv
    Untracked:  35_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T4-SRF_filtered_PSMs.xlsx
    Untracked:  36_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-BDCL_filtered_PSMs.tsv
    Untracked:  36_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  37_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-DCL_filtered_PSMs.tsv
    Untracked:  37_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-DCL_filtered_PSMs.xlsx
    Untracked:  38_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-SRF_filtered_PSMs.tsv
    Untracked:  38_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T5-SRF_filtered_PSMs.xlsx
    Untracked:  39_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-BDCL_filtered_PSMs.tsv
    Untracked:  39_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  40_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-DCL_filtered_PSMs.tsv
    Untracked:  40_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-DCL_filtered_PSMs.xlsx
    Untracked:  41_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-SRF_filtered_PSMs.tsv
    Untracked:  41_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T6-SRF_filtered_PSMs.xlsx
    Untracked:  42_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-BDCL_filtered_PSMs.tsv
    Untracked:  42_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-BDCL_filtered_PSMs.xlsx
    Untracked:  43_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-DCL_filtered_PSMs.tsv
    Untracked:  43_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-DCL_filtered_PSMs.xlsx
    Untracked:  44_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-SRF_filtered_PSMs.tsv
    Untracked:  44_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T7-SRF_filtered_PSMs.xlsx
    Untracked:  45_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-BDCL_filtered_PSMs.tsv
    Untracked:  45_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  46_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-DCL_filtered_PSMs.tsv
    Untracked:  46_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-DCL_filtered_PSMs.xlsx
    Untracked:  47_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-SRF_filtered_PSMs.tsv
    Untracked:  47_IPTL-CyanoSUMI-MI41M-18OMix-16O-B-T8-SRF_filtered_PSMs.xlsx
    Untracked:  48_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-BDCL_filtered_PSMs.tsv
    Untracked:  48_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-BDCL_filtered_PSMs.xlsx
    Untracked:  49_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-DCL_filtered_PSMs.tsv
    Untracked:  49_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-DCL_filtered_PSMs.xlsx
    Untracked:  50_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-SRF_filtered_PSMs.tsv
    Untracked:  50_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T1-SRF_filtered_PSMs.xlsx
    Untracked:  51_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-BDCL_filtered_PSMs.tsv
    Untracked:  51_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-BDCL_filtered_PSMs.xlsx
    Untracked:  52_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-DCL_filtered_PSMs.tsv
    Untracked:  52_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-DCL_filtered_PSMs.xlsx
    Untracked:  53_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-SRF_filtered_PSMs.tsv
    Untracked:  53_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T3-SRF_filtered_PSMs.xlsx
    Untracked:  54_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-BDCL_filtered_PSMs.tsv
    Untracked:  54_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-BDCL_filtered_PSMs.xlsx
    Untracked:  55_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-DCL_filtered_PSMs.tsv
    Untracked:  55_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-DCL_filtered_PSMs.xlsx
    Untracked:  56_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-SRF_filtered_PSMs.tsv
    Untracked:  56_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T4-SRF_filtered_PSMs.xlsx
    Untracked:  57_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-BDCL_filtered_PSMs.tsv
    Untracked:  57_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-BDCL_filtered_PSMs.xlsx
    Untracked:  58_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-DCL_filtered_PSMs.tsv
    Untracked:  58_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-DCL_filtered_PSMs.xlsx
    Untracked:  59_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-SRF_filtered_PSMs.tsv
    Untracked:  59_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T5-SRF_filtered_PSMs.xlsx
    Untracked:  60_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-BDCL_filtered_PSMs.tsv
    Untracked:  60_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-BDCL_filtered_PSMs.xlsx
    Untracked:  61_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-DCL_filtered_PSMs.tsv
    Untracked:  61_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-DCL_filtered_PSMs.xlsx
    Untracked:  62_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-SRF_filtered_PSMs.tsv
    Untracked:  62_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T6-SRF_filtered_PSMs.xlsx
    Untracked:  63_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-BDCL_filtered_PSMs.tsv
    Untracked:  63_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-BDCL_filtered_PSMs.xlsx
    Untracked:  64_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-DCL_filtered_PSMs.tsv
    Untracked:  64_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T7-DCL_filtered_PSMs.xlsx
    Untracked:  65_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-BDCL_filtered_PSMs.tsv
    Untracked:  65_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-BDCL_filtered_PSMs.xlsx
    Untracked:  66_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-DCL_filtered_PSMs.tsv
    Untracked:  66_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-DCL_filtered_PSMs.xlsx
    Untracked:  67_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-SRF_filtered_PSMs.tsv
    Untracked:  67_IPTL-CyanoSUMI-MI41M-18OMix-16O-C-T8-SRF_filtered_PSMs.xlsx
    Untracked:  Hist_freq_of_peptide_id.pdf
    Untracked:  Hist_intensity_replicate_centered.pdf
    Untracked:  Hist_intensity_replicate_not_centered.pdf
    Untracked:  Hist_intensity_spec_level_centered.pdf
    Untracked:  Hist_intensity_spec_level_not_centered.pdf
    Untracked:  Hist_number of timepoints_peptide_appears.pdf
    Untracked:  P1_number_unique_peptide_per_sample.pdf
    Untracked:  P2_number_of_cumulative_unique_peptides.pdf
    Untracked:  Peptide_id.tsv
    Untracked:  Replicate_ABC50_combined.tsv
    Untracked:  Replicate_ABC50_combined.xlsx
    Untracked:  Replicate_ABC63_combined.tsv
    Untracked:  Replicate_ABC63_combined.xlsx
    Untracked:  Replicate_ABC67_combined.tsv
    Untracked:  Replicate_ABC67_combined.xlsx
    Untracked:  Replicate_BC_combined.tsv
    Untracked:  Replicate_BC_combined.xlsx
    Untracked:  Replicate_B_combined.tsv
    Untracked:  Replicate_B_combined.xlsx
    Untracked:  Time_series_intesity_all.pdf
    Untracked:  data/Time_metadata.txt
    Untracked:  data/counts_example.csv
    Untracked:  data/diel-Feb21/
    Untracked:  data/diel-Feb25/
    Untracked:  data/diel-Feb26/
    Untracked:  data/diel-Mar2/
    Untracked:  data/synthetic_control.data.txt
    Untracked:  hclust_DTW_average_wide.pdf
    Untracked:  hclust_DTW_average_wide_after_normalization.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/peptide_level_visualization.Rmd</code>) and HTML (<code>docs/peptide_level_visualization.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/2ef4a7c7cd3bf243c13749e197280bb43c8b8256/analysis/peptide_level_visualization.Rmd" target="_blank">2ef4a7c</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
minor changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/cc8fcead0955c764959ec0c3e16e2a163294b0b1/docs/peptide_level_visualization.html" target="_blank">cc8fcea</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/71155f848db04d045425be78875e661717977684/analysis/peptide_level_visualization.Rmd" target="_blank">71155f8</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
splitting rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/3ca3689575a1ce177980d5e27a35bbdfd6c7457e/docs/peptide_level_visualization.html" target="_blank">3ca3689</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/3b32cf64c2bdcc63a2a6109fa1b697f4b95f59b8/analysis/peptide_level_visualization.Rmd" target="_blank">3b32cf6</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-10
</td>
<td>
important
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/8a7a3f8017e8c185df8cc6decf3fac69c1209c5e/docs/peptide_level_visualization.html" target="_blank">8a7a3f8</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/227f4ae68bd8936606faff4c78622c261dffb648/analysis/peptide_level_visualization.Rmd" target="_blank">227f4ae</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
spec_level
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/c974307139ff5f126397f0e41c958cd2f37eaaa9/docs/peptide_level_visualization.html" target="_blank">c974307</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/2a8d1baa0753a5fb1ce13c40005b1787bc4fbab4/analysis/peptide_level_visualization.Rmd" target="_blank">2a8d1ba</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
grid
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/f77b37f32a9ff55d40ef7d579a5cfc42f19c2c21/docs/peptide_level_visualization.html" target="_blank">f77b37f</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7f223a8f8561ef76901926468a4d13e2f6bcda7b/analysis/peptide_level_visualization.Rmd" target="_blank">7f223a8</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-04
</td>
<td>
intensity_histogram
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/3eb4d8892bf9cefac7d0cc85d46628e66e0de6b6/docs/peptide_level_visualization.html" target="_blank">3eb4d88</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/19d1e6c3242b4fd500c1500ad77e6ea298da142f/analysis/peptide_level_visualization.Rmd" target="_blank">19d1e6c</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
rain fdr
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/45127bd04e7e4abc5fa3898da9bbd1ddf3a56235/docs/peptide_level_visualization.html" target="_blank">45127bd</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/c6a81a1791ff2d152d4afefb1b87f0d47b4c97b6/analysis/peptide_level_visualization.Rmd" target="_blank">c6a81a1</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
x-axis label
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/cd5bef3583bb77b123c9e13ff671a87dcc3aa64c/docs/peptide_level_visualization.html" target="_blank">cd5bef3</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/21703f26decd3e16e0d1f5930a890f499b1ecada/analysis/peptide_level_visualization.Rmd" target="_blank">21703f2</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Rain
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/86aa80188d28226f3e412f84853d061b2187888a/docs/peptide_level_visualization.html" target="_blank">86aa801</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7a8405a4efd40a8494bdf38a3d224b56406bdbf5/analysis/peptide_level_visualization.Rmd" target="_blank">7a8405a</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
rain_not_done
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/403a8fb73c61d964f60bec6f048d4d6638c546c9/docs/peptide_level_visualization.html" target="_blank">403a8fb</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/2294f512531a7fb7d0fdf56e259e674cf6b6c570/analysis/peptide_level_visualization.Rmd" target="_blank">2294f51</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-03
</td>
<td>
clustering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/6a616230eede9f744ec3f3134d36affbb46ff1f3/docs/peptide_level_visualization.html" target="_blank">6a61623</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/5478476358f9190fb8438dfee193052b3c2e3bb0/analysis/peptide_level_visualization.Rmd" target="_blank">5478476</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
hclust
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/7218e38d19977eee4cdac9f49bb92a9c754c0687/docs/peptide_level_visualization.html" target="_blank">7218e38</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/dc0c92541c7a095cc906479e3d7db744c6e4f4ca/analysis/peptide_level_visualization.Rmd" target="_blank">dc0c925</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
average
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/52cb18451f2c50292d084abf3126d330ef6be2bb/docs/peptide_level_visualization.html" target="_blank">52cb184</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7f26b0f0ed5a088087ff67171af9bd22c65efe5f/analysis/peptide_level_visualization.Rmd" target="_blank">7f26b0f</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-03-02
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/c2f25d3613813ebed224e4e969c5753250e890fe/docs/peptide_level_visualization.html" target="_blank">c2f25d3</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/c5ec37ef6ebd29a83bdba8d242b57fccc7837df9/analysis/peptide_level_visualization.Rmd" target="_blank">c5ec37e</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-26
</td>
<td>
total 50 samples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/e6cbdfa41b1e0486cb84eb4cb9e37efb7dc3e5ff/docs/peptide_level_visualization.html" target="_blank">e6cbdfa</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/af94ad010f576c08f5207ee8e60e202cb93a8691/analysis/peptide_level_visualization.Rmd" target="_blank">af94ad0</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-25
</td>
<td>
peptide
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/3db29b0bc2c1673a15f5600c103841a88b5a47bf/docs/peptide_level_visualization.html" target="_blank">3db29b0</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/fe508d15413c128cce2d577485c74c32d77d1704/analysis/peptide_level_visualization.Rmd" target="_blank">fe508d1</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/e55579c873cb14ccd7bc7796c410ba177abc2a6c/docs/peptide_level_visualization.html" target="_blank">e55579c</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/44eb05fc0b315c8860224b5d2d2d40e0947e21dd/analysis/peptide_level_visualization.Rmd" target="_blank">44eb05f</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-23
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/1ca336372d560d548ec8e4f1ee97dd6bd9de4d26/docs/peptide_level_visualization.html" target="_blank">1ca3363</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/dd3354ea8f4f0fc98a1ac9e91aeb7b5acc3645c9/analysis/peptide_level_visualization.Rmd" target="_blank">dd3354e</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
peptide
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/69d6ce78a34ed33f4c1c6f84cc3d565cc46908a9/docs/peptide_level_visualization.html" target="_blank">69d6ce7</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/deb4db2f0aae05e8a37b3c9f238db1621ed35046/analysis/peptide_level_visualization.Rmd" target="_blank">deb4db2</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
peptide
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/6915013940ded02942cbecc91fd9b56b9ecbec28/docs/peptide_level_visualization.html" target="_blank">6915013</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/5a66cb1df60159b096f60da4df338e24228e996d/analysis/peptide_level_visualization.Rmd" target="_blank">5a66cb1</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
peptide
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/12b2c74add495b09b165c27e24022cdd0f863103/docs/peptide_level_visualization.html" target="_blank">12b2c74</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/c4cbcd229432d571f38ef352a2c584f61be1c6c1/docs/peptide_level_visualization.html" target="_blank">c4cbcd2</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/7830f0053fe892c583494bdfddfc272fe7ffbd2c/analysis/peptide_level_visualization.Rmd" target="_blank">7830f00</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/KiseokUchicago/BH_Diel/ee263ad22c2ca793bafd055ecf12f368d5b9386e/docs/peptide_level_visualization.html" target="_blank">ee263ad</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/KiseokUchicago/BH_Diel/blob/0ccca1470fe945598555adf482d6057be73b219f/analysis/peptide_level_visualization.Rmd" target="_blank">0ccca14</a>
</td>
<td>
KiseokUchicago
</td>
<td>
2021-02-22
</td>
<td>
visualization
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="peptide-level-visualization" class="section level3">
<h3>Peptide level visualization</h3>
<pre class="r"><code># libraries
library(dplyr)
library(tidyr)
library(stringr)
library(ggplot2)
library(formattable)</code></pre>
</div>
<div id="quality-check-number-of-uniq-peptides-sequence-per-sample" class="section level3">
<h3>1. Quality check: Number of uniq peptides (sequence) per sample</h3>
<pre class="r"><code># read in combined table of 24 samples (Replicate B)
df_pep &lt;- read.table(&quot;Replicate_ABC67_combined.tsv&quot;,sep=&#39;\t&#39;,header = T)

# count unique peptide per sample
df_uni &lt;- df_pep %&gt;% group_by(SampleID) %&gt;% summarize(uniq_pep = length(unique(Base.Peptide.Sequence)))
df_uni1 &lt;- df_uni %&gt;% separate(SampleID, sep=&#39;_&#39;, into= c(&#39;Depth&#39;,&#39;Time&#39;,&#39;Replicate&#39;),remove = F)
df_uni2 &lt;- df_uni1 %&gt;% arrange(desc(Depth))

# get the order right from SRF -&gt; DCL -&gt; BDCL
ord &lt;- df_uni2$SampleID 
df_uni2$SampleID &lt;- factor(df_uni2$SampleID, levels = ord)
df_uni2$Depth &lt;- factor(df_uni2$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))

# Making barplot
theme_set(theme_bw())
cols &lt;- c(&#39;SRF&#39;=&#39;#fc8d62&#39;,&#39;DCL&#39;=&#39;#66c2a5&#39;,&#39;BDCL&#39;=&#39;#8da0cb&#39;)
uniq1 &lt;- ggplot(df_uni2, aes(x=SampleID, fill=Depth, y=uniq_pep)) +  ## I put fill=value which was wrong
  geom_bar(stat=&quot;identity&quot;) +
  scale_fill_manual(values=cols) +
  ylab(&quot;Number of unique peptide sequence \n&quot;) +
  xlab(&quot;\n SampleID&quot;) +
  ggtitle(&quot;Number of unique peptide sequences per sample&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank())

uniq2 &lt;- uniq1 + geom_text(aes(x=df_uni$SampleID, y=df_uni$uniq_pep, label=df_uni$uniq_pep),
            size = 3, hjust = 0.5, position = position_stack(vjust = +1.05), family=&quot;serif&quot;)
uniq2</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-2-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>sample DCL_T8 doesn’t seem to have good quality MS data.   DCL_T8_A was added and now it seems okay.</p>
</div>
<div id="unique-number-of-peptides-per-timepoint" class="section level3">
<h3>1.1. Unique number of peptides per timepoint</h3>
<p>Are there any differences across time points?<br />
Nope.</p>
<pre class="r"><code># get # of unique peptides
df_uniq &lt;- df_pep %&gt;% group_by(Time,Depth) %&gt;% summarize(uniq_pep = length(unique(Base.Peptide.Sequence)))
df_uniq1 &lt;- df_uniq %&gt;% arrange(desc(Depth))

# get the order right from SRF -&gt; DCL -&gt; BDCL
df_uniq1$Depth &lt;- factor(df_uniq1$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))

# Making barplot
theme_set(theme_bw())
cols &lt;- c(&#39;SRF&#39;=&#39;#fc8d62&#39;,&#39;DCL&#39;=&#39;#66c2a5&#39;,&#39;BDCL&#39;=&#39;#8da0cb&#39;)
time1 &lt;- ggplot(df_uniq1, aes(x=Time, fill=Depth, y=uniq_pep)) +  ## I put fill=value which was wrong
  geom_bar(stat=&quot;identity&quot;,position=&quot;dodge&quot;) +
  scale_fill_manual(values=cols) +
  ylab(&quot;Number of unique peptide sequence \n&quot;) +
  xlab(&quot;\n Sampling Timepoint&quot;) +
  ggtitle(&quot;Number of unique peptide sequences per timepoint (including all replicates)&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))+
  theme(panel.grid.major = element_blank()) +
  theme(panel.grid.minor = element_blank())

time1</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-3-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="number-of-accumulative-uniq-peptides-sequence-does-it-saturate" class="section level3">
<h3>2. Number of accumulative uniq peptides (sequence): Does it saturate?</h3>
<pre class="r"><code># read in combined table of 24 samples (Replicate B)
df_pep &lt;- read.table(&quot;Replicate_ABC67_combined.tsv&quot;,sep=&#39;\t&#39;,header = T)

# get unique peptides per sample
dt &lt;- df_pep %&gt;% select(SampleID, Base.Peptide.Sequence) %&gt;% unique()
# change SampleID into number
dt1 &lt;- dt %&gt;% select(SampleID) %&gt;% unique() %&gt;% arrange(SampleID) %&gt;%  arrange()
dt2 &lt;- tibble::rownames_to_column(dt1,&quot;depth&quot;)
# merge dt and dt2
dt_depth &lt;- left_join(dt,dt2,by=c(&#39;SampleID&#39;=&#39;SampleID&#39;))
dim(dt_depth)</code></pre>
<pre><code>[1] 38075     3</code></pre>
<pre class="r"><code># get cumulative number of unique peptides (https://stackoverflow.com/questions/15698399/cumulative-count-of-unique-values-in-r)
require(data.table)
dt_depth &lt;- as.data.table(dt_depth)
setkey(dt_depth, &quot;depth&quot;)
dt_depth[, Base.Peptide.Sequence := as.numeric(factor(Base.Peptide.Sequence, levels = unique(Base.Peptide.Sequence)))]
setkey(dt_depth, &quot;depth&quot;, &quot;Base.Peptide.Sequence&quot;)
dt.out &lt;- dt_depth[J(unique(depth)), mult=&quot;last&quot;]
dt.out[, Base.Peptide.Sequence := cummax(Base.Peptide.Sequence)]

# getting ready to make plot
df.out &lt;- tibble::rowid_to_column(dt.out) 
df.out2 &lt;- df.out %&gt;% select(rowid, Base.Peptide.Sequence) %&gt;% rename(value=Base.Peptide.Sequence) ## 3366 unique peptides
# Making line plot
theme_set(theme_bw())

p1 &lt;- ggplot(df.out2,aes(x = rowid,y = value)) + geom_line(col=&quot;magenta&quot;) + geom_point(color=&#39;maroon&#39;) +
  ylab(&quot;Number of cumulative unique peptide sequence \n&quot;) +
  xlab(&quot;\n Number of cumulative samples&quot;) +
  # ylim(c(0,2200)) +
  ggtitle(&quot;Number of cumulative unique peptide sequences \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))
  # theme(panel.grid.major = element_blank())
  # theme(panel.grid.minor = element_blank())

p1</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-4-1.png" width="1056" style="display: block; margin: auto;" /> It seems to saturate as more samples are added.</p>
</div>
<div id="which-peptide_ids-are-present-in-allalmost-time-points" class="section level3">
<h3>3. Which Peptide_ids are present in all(almost) time points?</h3>
<p>Make a histogram of x-axis being the # of time points present, y-axis being the number of Peptide_ids<br />
We need 3 histogram for (1) Surface, (2) DCL, and (3) BDCL</p>
<pre class="r"><code># read in combined table of 24 samples (Replicate B)
df_pep &lt;- read.table(&quot;Replicate_ABC67_combined.tsv&quot;,sep=&#39;\t&#39;,header = T)

# (1) Surface
colnames(df_pep)</code></pre>
<pre><code> [1] &quot;Spec_id&quot;                 &quot;Base.Peptide.Sequence&quot;  
 [3] &quot;log2.16O.18O.Ratio&quot;      &quot;cent.log2.16O.18O.Ratio&quot;
 [5] &quot;SampleID&quot;                &quot;Index&quot;                  
 [7] &quot;Cruise&quot;                  &quot;Station&quot;                
 [9] &quot;Isotope&quot;                 &quot;Replicate&quot;              
[11] &quot;Time&quot;                    &quot;Depth&quot;                  
[13] &quot;SpectrumID&quot;              &quot;ScanID&quot;                 
[15] &quot;Target.&quot;                 &quot;Decoy.&quot;                 
[17] &quot;Peptide_id&quot;             </code></pre>
<pre class="r"><code>df_srf &lt;- df_pep %&gt;% filter(Depth == &#39;SRF&#39;) %&gt;% select(Peptide_id, SampleID, Time, Depth) %&gt;% unique() %&gt;% arrange(Peptide_id)
# How many time points did it appear
(df_srf2 &lt;- df_srf %&gt;% group_by(Peptide_id) %&gt;% summarize(Timepoints = n_distinct(Time)) %&gt;% ungroup())</code></pre>
<pre><code># A tibble: 2,081 x 2
   Peptide_id Timepoints
 * &lt;chr&gt;           &lt;int&gt;
 1 Pep_1               8
 2 Pep_10              8
 3 Pep_100             8
 4 Pep_1000            6
 5 Pep_1001            4
 6 Pep_1002            4
 7 Pep_1003            4
 8 Pep_1004            6
 9 Pep_1006            6
10 Pep_1007            4
# ... with 2,071 more rows</code></pre>
<pre class="r"><code># Check if Pep_1 appeared in 8 time points
df_srf %&gt;% filter(Peptide_id==&quot;Pep_1004&quot;) # true 8 times</code></pre>
<pre><code>  Peptide_id SampleID Time Depth
1   Pep_1004 SRF_T1_A   T1   SRF
2   Pep_1004 SRF_T2_A   T2   SRF
3   Pep_1004 SRF_T3_A   T3   SRF
4   Pep_1004 SRF_T4_A   T4   SRF
5   Pep_1004 SRF_T5_B   T5   SRF
6   Pep_1004 SRF_T7_B   T7   SRF</code></pre>
<pre class="r"><code># Histogram
theme_set(theme_bw())
ggplot(df_srf2, aes(Timepoints)) + geom_bar() +
  ylab(&quot;Number of peptides \n&quot;) +
  xlab(&quot;\n Cumulative number of timepoints peptide are present&quot;) +
  scale_x_continuous(breaks=seq(1,8,1))+
  ggtitle(&quot;Distribution of number of timepoints each Peptide_id appears (Surface) \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-5-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(panel.grid.major = element_blank())
  # theme(panel.grid.minor = element_blank())

# Let&#39;s try to plot all of them at once
df_all &lt;- df_pep %&gt;% select(Peptide_id, SampleID, Time, Depth) %&gt;% unique() %&gt;% arrange(Peptide_id)
# How many time points did it appear
(df_all2 &lt;- df_all %&gt;% group_by(Peptide_id, Depth) %&gt;% summarize(Timepoints = n_distinct(Time)) %&gt;% ungroup())</code></pre>
<pre><code># A tibble: 6,104 x 3
   Peptide_id Depth Timepoints
   &lt;chr&gt;      &lt;chr&gt;      &lt;int&gt;
 1 Pep_1      BDCL           8
 2 Pep_1      DCL            8
 3 Pep_1      SRF            8
 4 Pep_10     BDCL           8
 5 Pep_10     DCL            8
 6 Pep_10     SRF            8
 7 Pep_100    BDCL           8
 8 Pep_100    DCL            8
 9 Pep_100    SRF            8
10 Pep_1000   BDCL           1
# ... with 6,094 more rows</code></pre>
<pre class="r"><code>df_all2$Depth &lt;- factor(df_all2$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))

# Histogram
ggplot(df_all2, aes(Timepoints)) + geom_bar(aes(fill=Depth), position=&#39;dodge&#39;) +
  scale_fill_manual(values=cols) +
  ylab(&quot;Number of peptides \n&quot;) +
  xlab(&quot;\n Cumulative number of timepoints peptide are present&quot;) +
  scale_x_continuous(breaks=seq(1,8,1))+
  ggtitle(&quot;Distribution of number of timepoints each Peptide_id appears \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-5-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(panel.grid.major = element_blank())
  # theme(panel.grid.minor = element_blank())</code></pre>
</div>
<div id="for-the-peptides-that-show-up-only-in-1-time-point" class="section level3">
<h3>3.2. For the peptides that show up only in 1 time point</h3>
<pre class="r"><code># filter peptides that appear only once
df_once &lt;- df_all2 %&gt;% filter(Timepoints ==1)
# innerjoin to get only the peptide_id from above (that appear once)
df_norep &lt;- df_all %&gt;% select(-SampleID) %&gt;% unique() # get unique peptide_id per Sample (not replicate)
df_once2 &lt;- inner_join(df_norep,df_once,by=c(&quot;Depth&quot;=&quot;Depth&quot;,&quot;Peptide_id&quot;=&quot;Peptide_id&quot;))
df_once2$Depth &lt;- factor(df_once2$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))


# Histogram
theme_set(theme_bw())
ggplot(df_once2, aes(Time)) + geom_bar(aes(fill=Depth), position=&#39;dodge&#39;) +
  scale_fill_manual(values=cols) +
  ylab(&quot;Number of unique peptides \n&quot;) +
  xlab(&quot;\n Sampling Timepoint&quot;) +
  # scale_x_continuous(breaks=seq(1,8,1))+
  ggtitle(&quot;For peptide_ids detected once, in what timepoint is it detected? \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-6-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(panel.grid.major = element_blank())
  # theme(panel.grid.minor = element_blank())</code></pre>
<p>In the surface and deep chlorophyll layer at T1 and T7, the numbers seem to surge.<br />
This is when the sun is at its peak.</p>
</div>
<div id="plotting-time-series-fluctuation-for-the-peptides" class="section level3">
<h3>4. Plotting time series fluctuation for the peptides</h3>
<p>First retrieve the log abundance_ratio value for all peptides (average over the same spec_id)</p>
<pre class="r"><code># (1) Get list of peptides that are present in 7 or 8 timepoints
df_all2_filt &lt;- df_all2 %&gt;% filter(Timepoints &gt; 6) %&gt;% unique() 

# (2) Let&#39;s average the abundance_ratio for each unique peptide
df_int &lt;- df_pep %&gt;% select(Peptide_id, Spec_id, cent.log2.16O.18O.Ratio, SampleID, Time, Depth, Replicate) %&gt;%
  unique() %&gt;% arrange(Peptide_id)
head(df_int)</code></pre>
<pre><code>  Peptide_id         Spec_id cent.log2.16O.18O.Ratio  SampleID Time Depth
1      Pep_1 BDCL_T1_A_31195              -0.3768094 BDCL_T1_A   T1  BDCL
2      Pep_1 BDCL_T1_A_31179              -0.2659891 BDCL_T1_A   T1  BDCL
3      Pep_1 BDCL_T1_A_31003              -0.4227962 BDCL_T1_A   T1  BDCL
4      Pep_1 BDCL_T1_A_31022              -0.3675320 BDCL_T1_A   T1  BDCL
5      Pep_1 BDCL_T1_A_31372              -0.3737250 BDCL_T1_A   T1  BDCL
6      Pep_1 BDCL_T1_A_31355              -0.4533029 BDCL_T1_A   T1  BDCL
  Replicate
1         A
2         A
3         A
4         A
5         A
6         A</code></pre>
<pre class="r"><code>df_int %&gt;% filter(Peptide_id==&#39;Pep_1&#39; &amp; Time ==&#39;T1&#39; &amp; SampleID == &#39;BDCL_T1_A&#39;)</code></pre>
<pre><code>   Peptide_id         Spec_id cent.log2.16O.18O.Ratio  SampleID Time Depth
1       Pep_1 BDCL_T1_A_31195              -0.3768094 BDCL_T1_A   T1  BDCL
2       Pep_1 BDCL_T1_A_31179              -0.2659891 BDCL_T1_A   T1  BDCL
3       Pep_1 BDCL_T1_A_31003              -0.4227962 BDCL_T1_A   T1  BDCL
4       Pep_1 BDCL_T1_A_31022              -0.3675320 BDCL_T1_A   T1  BDCL
5       Pep_1 BDCL_T1_A_31372              -0.3737250 BDCL_T1_A   T1  BDCL
6       Pep_1 BDCL_T1_A_31355              -0.4533029 BDCL_T1_A   T1  BDCL
7       Pep_1 BDCL_T1_A_30845               0.1603187 BDCL_T1_A   T1  BDCL
8       Pep_1 BDCL_T1_A_30832               0.2822996 BDCL_T1_A   T1  BDCL
9       Pep_1 BDCL_T1_A_31549              -0.2180393 BDCL_T1_A   T1  BDCL
10      Pep_1 BDCL_T1_A_31536              -0.1778131 BDCL_T1_A   T1  BDCL
11      Pep_1 BDCL_T1_A_31789              -0.3197409 BDCL_T1_A   T1  BDCL
12      Pep_1 BDCL_T1_A_32226              -0.4774925 BDCL_T1_A   T1  BDCL
   Replicate
1          A
2          A
3          A
4          A
5          A
6          A
7          A
8          A
9          A
10         A
11         A
12         A</code></pre>
<pre class="r"><code># first average the spectrum_id per sample 
# important to make sure to take the mean of the log raio (instead of taking log2 after averaging the original value(2^x))
df_ave &lt;- df_int %&gt;% group_by(SampleID, Time, Depth, Replicate, Peptide_id) %&gt;%
  summarize(Mean_abundance_ratio=mean(cent.log2.16O.18O.Ratio),
            SD_pep=sd(cent.log2.16O.18O.Ratio)) %&gt;% ungroup()
head(df_ave)</code></pre>
<pre><code># A tibble: 6 x 7
  SampleID  Time  Depth Replicate Peptide_id Mean_abundance_ratio  SD_pep
  &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;                     &lt;dbl&gt;   &lt;dbl&gt;
1 BDCL_T1_A T1    BDCL  A         Pep_1                   -0.251   0.240 
2 BDCL_T1_A T1    BDCL  A         Pep_10                   0.229   0.241 
3 BDCL_T1_A T1    BDCL  A         Pep_100                  0.0658  0.286 
4 BDCL_T1_A T1    BDCL  A         Pep_101                  0.0175  0.0496
5 BDCL_T1_A T1    BDCL  A         Pep_1014                 0.863  NA     
6 BDCL_T1_A T1    BDCL  A         Pep_1015                 0.158   0.0991</code></pre>
<pre class="r"><code>dim(df_ave)</code></pre>
<pre><code>[1] 38075     7</code></pre>
<pre class="r"><code>df_ave %&gt;% filter(Peptide_id==&#39;Pep_1&#39; &amp; Time ==&#39;T1&#39;,Depth==&quot;BDCL&quot;) # check</code></pre>
<pre><code># A tibble: 3 x 7
  SampleID  Time  Depth Replicate Peptide_id Mean_abundance_ratio SD_pep
  &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;                     &lt;dbl&gt;  &lt;dbl&gt;
1 BDCL_T1_A T1    BDCL  A         Pep_1                   -0.251   0.240
2 BDCL_T1_B T1    BDCL  B         Pep_1                   -0.0163  0.212
3 BDCL_T1_C T1    BDCL  C         Pep_1                    0.506   0.277</code></pre>
<pre class="r"><code># Now, average and error by replicate
# see here for calculating error bar in log scale (https://faculty.washington.edu/stuve/log_error.pdf)
# but here we will not use the method in the link above
# calculate average and error bar for abundance_ratio
# SE calculated with Jake Waldbauer&#39;s method: sqrt(se(Mean_abundance_ratio)^2 +(mean(SD_pep))^2))
# standard error
se &lt;- function(x) sd(x)/sqrt(length(x))
df_ave_rep &lt;- df_ave %&gt;% group_by(Time, Depth, Peptide_id) %&gt;%
  summarize(Ave_abundance_ratio=mean(Mean_abundance_ratio),
            SE_abundance_ratio=sqrt((se(Mean_abundance_ratio))^2 +(mean(SD_pep))^2)) %&gt;% ungroup()
head(df_ave_rep,10)</code></pre>
<pre><code># A tibble: 10 x 5
   Time  Depth Peptide_id Ave_abundance_ratio SE_abundance_ratio
   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;                    &lt;dbl&gt;              &lt;dbl&gt;
 1 T1    BDCL  Pep_1                   0.0796              0.330
 2 T1    BDCL  Pep_10                 -0.185               0.270
 3 T1    BDCL  Pep_100                 0.214               0.149
 4 T1    BDCL  Pep_1003                0.0789             NA    
 5 T1    BDCL  Pep_1007                0.244              NA    
 6 T1    BDCL  Pep_1008               -0.340              NA    
 7 T1    BDCL  Pep_101                 0.191               0.191
 8 T1    BDCL  Pep_1011                0.253              NA    
 9 T1    BDCL  Pep_1012               -0.267              NA    
10 T1    BDCL  Pep_1014                0.863              NA    </code></pre>
<pre class="r"><code># why is there no sd for Pep_1004?
df_ave %&gt;% group_by(SampleID) %&gt;% filter(Peptide_id==&#39;Pep_1003&#39;) # there is only 1 sample for T1 BDCL </code></pre>
<pre><code># A tibble: 10 x 7
# Groups:   SampleID [10]
   SampleID  Time  Depth Replicate Peptide_id Mean_abundance_ratio   SD_pep
   &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;                     &lt;dbl&gt;    &lt;dbl&gt;
 1 BDCL_T1_B T1    BDCL  B         Pep_1003                 0.0789 NA      
 2 BDCL_T7_B T7    BDCL  B         Pep_1003                -0.0795 NA      
 3 DCL_T3_A  T3    DCL   A         Pep_1003                 0.248  NA      
 4 SRF_T1_A  T1    SRF   A         Pep_1003                 0.293   0.125  
 5 SRF_T1_B  T1    SRF   B         Pep_1003                -0.102  NA      
 6 SRF_T2_A  T2    SRF   A         Pep_1003                 0.308  NA      
 7 SRF_T3_A  T3    SRF   A         Pep_1003                 0.288   0.00194
 8 SRF_T3_B  T3    SRF   B         Pep_1003                 0.0473  0.138  
 9 SRF_T4_A  T4    SRF   A         Pep_1003                 0.205   0.0673 
10 SRF_T4_B  T4    SRF   B         Pep_1003                 0.0444  0.196  </code></pre>
<pre class="r"><code>df_ave_rep2 &lt;- df_ave_rep %&gt;% mutate(error_up=Ave_abundance_ratio+SE_abundance_ratio) %&gt;%
  mutate(error_down=Ave_abundance_ratio-SE_abundance_ratio)

head(df_ave_rep2)</code></pre>
<pre><code># A tibble: 6 x 7
  Time  Depth Peptide_id Ave_abundance_rat~ SE_abundance_ra~ error_up error_down
  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;                   &lt;dbl&gt;            &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;
1 T1    BDCL  Pep_1                  0.0796            0.330   0.410     -0.251 
2 T1    BDCL  Pep_10                -0.185             0.270   0.0858    -0.455 
3 T1    BDCL  Pep_100                0.214             0.149   0.363      0.0658
4 T1    BDCL  Pep_1003               0.0789           NA      NA         NA     
5 T1    BDCL  Pep_1007               0.244            NA      NA         NA     
6 T1    BDCL  Pep_1008              -0.340            NA      NA         NA     </code></pre>
<pre class="r"><code># (3) import time metadata and merge
(df_time &lt;- read.table(&#39;data/Time_metadata.txt&#39;, header=T, sep=&#39;\t&#39;))</code></pre>
<pre><code>  timepoint          date_time elapsed_hours
1        T1 6/26/2019 12:45:00             0
2        T2 6/26/2019 15:45:00             3
3        T3 6/26/2019 18:45:00             6
4        T4 6/26/2019 21:45:00             9
5        T5  6/27/2019 4:40:00            15
6        T6  6/27/2019 7:45:00            19
7        T7 6/27/2019 10:45:00            22
8        T8 6/27/2019 13:45:00            25</code></pre>
<pre class="r"><code>df_time$date_time &lt;- str_replace(df_time$date_time, &quot;/2019&quot;,&quot; &quot;)
df_time$date_time &lt;- str_replace(df_time$date_time, &quot;:00$&quot;,&quot;&quot;)

# left_join with average data
df_ave2 &lt;- df_ave_rep2 %&gt;% left_join(df_time, by=c(&#39;Time&#39;=&#39;timepoint&#39;))
head(df_ave2)</code></pre>
<pre><code># A tibble: 6 x 9
  Time  Depth Peptide_id Ave_abundance_rat~ SE_abundance_ra~ error_up error_down
  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;                   &lt;dbl&gt;            &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;
1 T1    BDCL  Pep_1                  0.0796            0.330   0.410     -0.251 
2 T1    BDCL  Pep_10                -0.185             0.270   0.0858    -0.455 
3 T1    BDCL  Pep_100                0.214             0.149   0.363      0.0658
4 T1    BDCL  Pep_1003               0.0789           NA      NA         NA     
5 T1    BDCL  Pep_1007               0.244            NA      NA         NA     
6 T1    BDCL  Pep_1008              -0.340            NA      NA         NA     
# ... with 2 more variables: date_time &lt;chr&gt;, elapsed_hours &lt;int&gt;</code></pre>
<pre class="r"><code># only use peptides that appear 7 or 8 times
df_ave2 %&gt;% select(Peptide_id) %&gt;% unique() %&gt;% dim() # 3233 peptides before</code></pre>
<pre><code>[1] 3366    1</code></pre>
<pre class="r"><code>df_filt78 &lt;- inner_join(df_ave2, df_all2_filt, by=c(&#39;Peptide_id&#39;=&#39;Peptide_id&#39;,&#39;Depth&#39;=&#39;Depth&#39;))

# inspecting if the inner join was successful
head(df_filt78)</code></pre>
<pre><code># A tibble: 6 x 10
  Time  Depth Peptide_id Ave_abundance_rat~ SE_abundance_ra~ error_up error_down
  &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;                   &lt;dbl&gt;            &lt;dbl&gt;    &lt;dbl&gt;      &lt;dbl&gt;
1 T1    BDCL  Pep_1                  0.0796            0.330   0.410   -0.251   
2 T1    BDCL  Pep_10                -0.185             0.270   0.0858  -0.455   
3 T1    BDCL  Pep_100                0.214             0.149   0.363    0.0658  
4 T1    BDCL  Pep_101                0.191             0.191   0.382   -0.000815
5 T1    BDCL  Pep_102                0.272            NA      NA       NA       
6 T1    BDCL  Pep_103                0.478             0.176   0.654    0.302   
# ... with 3 more variables: date_time &lt;chr&gt;, elapsed_hours &lt;int&gt;,
#   Timepoints &lt;int&gt;</code></pre>
<pre class="r"><code>colnames(df_ave2)</code></pre>
<pre><code>[1] &quot;Time&quot;                &quot;Depth&quot;               &quot;Peptide_id&quot;         
[4] &quot;Ave_abundance_ratio&quot; &quot;SE_abundance_ratio&quot;  &quot;error_up&quot;           
[7] &quot;error_down&quot;          &quot;date_time&quot;           &quot;elapsed_hours&quot;      </code></pre>
<pre class="r"><code>colnames(df_time)</code></pre>
<pre><code>[1] &quot;timepoint&quot;     &quot;date_time&quot;     &quot;elapsed_hours&quot;</code></pre>
<pre class="r"><code>colnames(df_filt78)</code></pre>
<pre><code> [1] &quot;Time&quot;                &quot;Depth&quot;               &quot;Peptide_id&quot;         
 [4] &quot;Ave_abundance_ratio&quot; &quot;SE_abundance_ratio&quot;  &quot;error_up&quot;           
 [7] &quot;error_down&quot;          &quot;date_time&quot;           &quot;elapsed_hours&quot;      
[10] &quot;Timepoints&quot;         </code></pre>
<pre class="r"><code>df_filt78 %&gt;% select(Peptide_id) %&gt;% unique() %&gt;% dim() # 841 peptides for peptides in 7 or 8 timepoints</code></pre>
<pre><code>[1] 841   1</code></pre>
<pre class="r"><code>df_filt78$Depth &lt;- factor(df_filt78$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))

# get only 8 ones
df_filt8 &lt;- df_filt78 %&gt;% filter(Timepoints &gt; 7)
df_filt8 %&gt;% select(Peptide_id) %&gt;% unique() %&gt;% dim() # 661 peptides for peptides in all 8 timepoints</code></pre>
<pre><code>[1] 661   1</code></pre>
<pre class="r"><code>df_filt8$Depth &lt;- factor(df_filt8$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))

# get only 7 ones
df_filt7 &lt;- df_filt78 %&gt;% filter(Timepoints == 7)
df_filt7 %&gt;% select(Peptide_id) %&gt;% unique() %&gt;% dim() # 356 peptides for peptides in 7 timepoints</code></pre>
<pre><code>[1] 356   1</code></pre>
<pre class="r"><code>df_filt7$Depth &lt;- factor(df_filt7$Depth, levels = c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;))</code></pre>
<p>Now I do it differently: just + and - the standard error. This is how I used to derive the error bar before (<a href="https://faculty.washington.edu/stuve/log_error.pdf" class="uri">https://faculty.washington.edu/stuve/log_error.pdf</a>) log scale error bar: d(log2y)=1/log2 * (dy / y) df_ave_rep2 &lt;- df_ave_rep %&gt;% mutate(ave_abundance_ratio = 2^Ave_abundance_ratio) %&gt;% mutate(rel_error = sd_abundance_ratio/ave_abundance_ratio) %&gt;% mutate(log2_sd_abundance_ratio = (1/log(2))*rel_error) %&gt;% mutate(error_up=Ave_abundance_ratio+log2_sd_abundance_ratio) %&gt;% mutate(error_down=Ave_abundance_ratio-log2_sd_abundance_ratio)</p>
<p>Plot for peptides present in 7 or 8 timepoints</p>
<pre class="r"><code># (4) Plot the scatter plot with error bars
theme_set(theme_bw())

ggplot(df_filt78, aes(x=elapsed_hours,y=Ave_abundance_ratio, color=Depth, group=interaction(Peptide_id,Depth))) +
  geom_point(size=2, shape=21) +
  geom_line()+
  geom_errorbar(aes(ymin=error_down, ymax=error_up), width=.2, position=position_dodge(.9))+
  scale_color_manual(values=cols) +
  ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
  xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
  scale_x_continuous(breaks=df_time$elapsed_hours,labels =c(df_time$date_time))+
  ggtitle(&quot;Time series abundance_ratio of peptide_id \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10)) +
  annotate(&quot;rect&quot;, xmin = 7.75, xmax = 16.53, ymin = -Inf, ymax = Inf, alpha = .2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-8-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # geom_rect(aes(xmin=7.75, xmax=16.53, ymin=0, ymax=Inf))
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank())</code></pre>
<p>without error bars</p>
<pre class="r"><code>theme_set(theme_bw())
ggplot(df_filt78, aes(x=elapsed_hours,y=Ave_abundance_ratio, color=Depth, group=interaction(Peptide_id,Depth))) +
  geom_point(size=2, shape=21) +
  geom_line()+
  # geom_errorbar(aes(ymin=error_down, ymax=error_up), width=.2, position=position_dodge(.9))+
  scale_color_manual(values=cols) +
  ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
  xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
  scale_x_continuous(breaks=df_time$elapsed_hours,labels =c(df_time$date_time))+
  ggtitle(&quot;Time series abundance_ratio of peptide_id \n&quot;)+
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10)) +
  annotate(&quot;rect&quot;, xmin = 7.75, xmax = 16.53, ymin = -Inf, ymax = Inf, alpha = .2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-9-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="surfacesrf" class="section level3">
<h3>4.1. Surface(SRF)</h3>
<pre class="r"><code># color code
# 1. Surface(orange): #fc8d62
# 2. DCL(green): #66c2a5
# 3. BDCL(blue): #8da0cb
color_vec &lt;- c(&#39;#fc8d62&#39;,&#39;#66c2a5&#39;,&#39;#8da0cb&#39;)
depth_vec &lt;- c(&#39;SRF&#39;,&#39;DCL&#39;,&#39;BDCL&#39;)

time_series_1depth_woError &lt;- function(df_filt78, depth){
  theme_set(theme_bw())
  # filter depth
  df_filt78_s &lt;- df_filt78 %&gt;% filter(Depth == depth_vec[depth])
  ggplot(df_filt78_s, aes(x=elapsed_hours,y=Ave_abundance_ratio, group=Peptide_id)) +
    geom_point(size=2, shape=21, color=color_vec[depth]) +
    geom_line(color= color_vec[depth])+
    ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
    xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
    scale_x_continuous(breaks=df_time$elapsed_hours,labels =c(df_time$date_time))+
    ggtitle(paste0(&quot;Time series abundance_ratio of peptide_id&quot;,&quot; (&quot;,depth_vec[depth],&quot;)&quot;,&quot;\n&quot;))+
    ## adjust positions
    theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
    theme(axis.text.y = element_text(size=10)) +
    annotate(&quot;rect&quot;, xmin = 7.75, xmax = 16.53, ymin = -Inf, ymax = Inf, alpha = .2)
    # theme(panel.grid.major = element_blank()) +
    # theme(panel.grid.minor = element_blank())
}

time_series_1depth_wError &lt;- function(df_filt78, depth){
  theme_set(theme_bw())
  # filter depth
  df_filt78_s &lt;- df_filt78 %&gt;% filter(Depth == depth_vec[depth])
  ggplot(df_filt78_s, aes(x=elapsed_hours,y=Ave_abundance_ratio, group=Peptide_id)) +
    geom_point(size=2, shape=21, color=color_vec[depth]) +
    geom_line(color= color_vec[depth])+
    geom_errorbar(aes(ymin=error_down, ymax=error_up), width=.05, position=position_dodge(.2),color=&#39;red&#39;)+
    ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
    xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
    scale_x_continuous(breaks=df_time$elapsed_hours,labels =c(df_time$date_time))+
    ggtitle(paste0(&quot;Time series abundance_ratio of peptide_id&quot;,&quot; (&quot;,depth_vec[depth],&quot;)&quot;,&quot;\n&quot;))+
    ## adjust positions
    theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
    theme(axis.text.y = element_text(size=10)) +
    annotate(&quot;rect&quot;, xmin = 7.75, xmax = 16.53, ymin = -Inf, ymax = Inf, alpha = .2)
    # theme(panel.grid.major = element_blank()) +
    # theme(panel.grid.minor = element_blank())
}

# when we use all 7, 8 appearance cases
time_series_1depth_woError(df_filt78,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt78,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 8 appearance cases
time_series_1depth_woError(df_filt8,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt8,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 7 appearance cases
time_series_1depth_woError(df_filt7,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt7,1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-10-6.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="deep-chlorophyll-layer-dcl" class="section level3">
<h3>4.2. Deep Chlorophyll layer (DCL)</h3>
<pre class="r"><code># when we use all 7, 8 appearance cases
time_series_1depth_woError(df_filt78,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt78,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 8 appearance cases
time_series_1depth_woError(df_filt8,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt8,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 7 appearance cases
time_series_1depth_woError(df_filt7,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt7,2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-11-6.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="below-deep-chlorophyll-layer-bdcl" class="section level3">
<h3>4.3. Below Deep Chlorophyll layer (BDCL)</h3>
<pre class="r"><code># when we use all 7, 8 appearance cases
time_series_1depth_woError(df_filt78,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt78,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 8 appearance cases
time_series_1depth_woError(df_filt8,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt8,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># when we use only 7 appearance cases
time_series_1depth_woError(df_filt7,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_1depth_wError(df_filt7,3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-12-6.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="ordination-analysis" class="section level3">
<h3>5. Ordination analysis</h3>
<p>For time series clustering with R, the first step is to work out an appropriate distance/similarity metric, and then, at the second step, use existing clustering techniques, such as k-means, hierarchical clustering, density-based clustering or subspace clustering, to find clustering structures. (<a href="http://www.rdatamining.com/examples/time-series-clustering-classification" class="uri">http://www.rdatamining.com/examples/time-series-clustering-classification</a>)</p>
</div>
<div id="hierarchical-clustering-using-dynamic-time-warping-distance" class="section level3">
<h3>5.1. Hierarchical clustering using Dynamic time warping distance</h3>
<p>Dynamic Time Warping (DTW) finds optimal alignment between two time series, and DTW distance is used as a distance metric in the example below. (<a href="http://www.rdatamining.com/examples/time-series-clustering-classification" class="uri">http://www.rdatamining.com/examples/time-series-clustering-classification</a>)</p>
<pre class="r"><code># let&#39;s try cluster our time series data
colnames(df_filt8)</code></pre>
<pre><code> [1] &quot;Time&quot;                &quot;Depth&quot;               &quot;Peptide_id&quot;         
 [4] &quot;Ave_abundance_ratio&quot; &quot;SE_abundance_ratio&quot;  &quot;error_up&quot;           
 [7] &quot;error_down&quot;          &quot;date_time&quot;           &quot;elapsed_hours&quot;      
[10] &quot;Timepoints&quot;         </code></pre>
<pre class="r"><code>df_8 &lt;- df_filt8 %&gt;% filter(Depth==&#39;SRF&#39;) %&gt;% select(Peptide_id,Time,Ave_abundance_ratio)

# reshape the dataframe with timepoints
library(maditr)
v_names &lt;- c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;)
df_8$Time &lt;- factor(df_8$Time, levels=v_names)
dc_8 &lt;- dcast(df_8, Peptide_id ~ Time, value.var = &quot;Ave_abundance_ratio&quot;)
# make Peptide_id more simpler (Pep_1 -&gt; 1)
dc_8$Peptide_id &lt;- str_replace(dc_8$Peptide_id,&quot;Pep_&quot;,&quot;P.&quot;)
# make first column into row name
dc_8 &lt;- tibble::column_to_rownames(dc_8, var=&quot;Peptide_id&quot;)

# DTW distances
library(dtw)
any(is.na(dc_8)) # False: there is no NA in the table</code></pre>
<pre><code>[1] FALSE</code></pre>
<pre class="r"><code>distMatrix &lt;- dist(dc_8, method=&quot;DTW&quot;)
hc &lt;- hclust(distMatrix, method=&quot;average&quot;)
plot(hc, cex=0.7, xlab=&#39;Hierarchical clustering of Distance(DTW) Matrix&#39;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-13-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Draw the particular small cluster in the time series plot</p>
<pre class="r"><code># function for plotting cluster of piptides
time_series_clust_wError &lt;- function(df_filt78, depth, clust1){
  theme_set(theme_bw())
  # filter depth
  df_filt78_s &lt;- df_filt78 %&gt;% filter(Depth == depth_vec[depth]) %&gt;% filter(Peptide_id %in% clust1)
  ggplot(df_filt78_s, aes(x=elapsed_hours,y=Ave_abundance_ratio, group=Peptide_id)) +
    geom_point(size=2, shape=21, color=color_vec[depth]) +
    geom_line(color= color_vec[depth])+
    geom_errorbar(aes(ymin=error_down, ymax=error_up), width=.05, position=position_dodge(.2),color=&#39;red&#39;)+
    ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
    xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
    scale_x_continuous(breaks=df_time$elapsed_hours,labels =c(df_time$date_time))+
    ggtitle(paste0(&quot;Time series abundance_ratio of peptide_id&quot;,&quot; (&quot;,depth_vec[depth],&quot;)&quot;,&quot;\n&quot;))+
    ## adjust positions
    theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
    theme(axis.text.y = element_text(size=10)) +
    annotate(&quot;rect&quot;, xmin = 7.75, xmax = 16.53, ymin = -Inf, ymax = Inf, alpha = .2)
    # theme(panel.grid.major = element_blank()) +
    # theme(panel.grid.minor = element_blank())
}

# cluster1 (outgroup-like)
clust1 &lt;- c(&#39;632&#39;,&quot;596&quot;,&quot;443&quot;,&quot;453&quot;,&quot;522&quot;,&quot;153&quot;,&quot;851&quot;,&quot;493&quot;,&quot;50&quot;,&quot;574&quot;,&quot;584&quot;,&quot;402&quot;,&quot;813&quot;)
clust1 &lt;- str_replace(clust1,&quot;^&quot;,&quot;Pep_&quot;) # make Peptide_id
time_series_clust_wError(df_filt8, 1, clust1)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-14-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># cluster2
clust2 &lt;- c(&quot;197&quot;,&quot;78&quot;,&quot;562&quot;,&quot;112&quot;,&quot;87&quot;,&quot;310&quot;,&quot;207&quot;, &quot;47&quot;,&quot;319&quot;,&quot;91&quot;,&quot;289&quot;,&quot;82&quot;,&quot;697&quot;,&quot;441&quot;,&quot;142&quot;,&quot;191&quot;, &quot;34&quot;,&quot;105&quot;,&quot;426&quot;,&quot;882&quot;,&quot;8&quot;,&quot;18&quot;,&quot;5&quot;,&quot;132&quot;,&quot;327&quot;)
clust2 &lt;- str_replace(clust2,&quot;^&quot;,&quot;Pep_&quot;) # make Peptide_id
time_series_clust_wError(df_filt8, 1, clust2)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-14-2.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="principal-coordinate-analysis-ordination-plot-pcoa-with-dtw-distance" class="section level3">
<h3>5.2. Principal coordinate analysis (ordination plot, PCoA) with DTW distance</h3>
<p>Use the DTW distance matrix to plot PCoA</p>
<pre class="r"><code># Use the 
# Transform distMatrix into matrix A
distmat &lt;- as(distMatrix, &quot;matrix&quot;)
dim(distmat)</code></pre>
<pre><code>[1] 532 532</code></pre>
<pre class="r"><code>A &lt;- -1/2*distmat^2
dim(A)</code></pre>
<pre><code>[1] 532 532</code></pre>
<pre class="r"><code># Scaling/centering of Matrix A
RowM &lt;- rowMeans(A)
ColM &lt;- colMeans(A)
M &lt;- mean(A)
Delta &lt;- A
for (i in 1:dim(A)[1]){
        for (j in 1:dim(A)[2]) {
                Delta[i,j] &lt;- Delta[i,j] - RowM[i] - ColM[j] + M
        }
}

# Compute the eigenvalues and eigenvectors of Delta
Eigen &lt;- eigen(Delta)
# SEE YOU GET 5 EIGENVALUES (nrow = ncol of sqare distance matrix)
# Scale the eigenvectors
FirstEigenvector &lt;- Eigen$vectors[,1]
FirstEigenvalue &lt;- Eigen$values[1]
SecondEigenvector &lt;- Eigen$vectors[,2]
SecondEigenvalue &lt;- Eigen$values[2]

PrincipalCoords &lt;- data.frame(First = FirstEigenvector*sqrt(FirstEigenvalue),
                              Sedond = SecondEigenvector * sqrt(SecondEigenvalue))

# we can take a shortcut instead: cmdScale or pco is now equal your PrincipalCoords (https://rstudio-pubs-static.s3.amazonaws.com/68544_06343669257d4f35aaca449f9ff1e6f7.html, https://genomicsclass.github.io/book/pages/mds.html)
require(labdsv)
pco_pep &lt;- pco(distmat, k =2)
# pco_pep2 &lt;- cmdscale(distmat, k =2)
# cmdscale(distmat, 2, eig=TRUE)$GOF

# plot PCoA without peptide_id labels
ggplot(as.data.frame(pco_pep$points), aes(x = V1, y = V2, label = row.names(as.data.frame(pco_pep$points)))) + 
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 4, color=&#39;pink&#39;,alpha=0.7)+
  xlab(paste0(&#39;\n PCoA1&#39;, &quot; (explained variance: &quot;,round(pco_pep$GOF[1],3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PCoA2&quot;,&quot; (explained variance: &quot;,round(pco_pep$GOF[2]-pco_pep$GOF[1],3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Peptide ordination PCoA plot (DTW distance) \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.y = element_text(size = 13,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
  theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-15-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot PCoA with peptide_id labels
library(ggrepel)
PCoA_1 &lt;- ggplot(as.data.frame(pco_pep$points), aes(x = V1, y = V2, label = row.names(as.data.frame(pco_pep$points)))) + 
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 4, color=&#39;pink&#39;,alpha=0.7)+
  xlab(paste0(&#39;\n PCoA1&#39;, &quot; (explained variance: &quot;,round(pco_pep$GOF[1],3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PCoA2&quot;,&quot; (explained variance: &quot;,round(pco_pep$GOF[2]-pco_pep$GOF[1],3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Peptide ordination PCoA plot (DTW distance) \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
  theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;)) +
  ggrepel::geom_label_repel()
  # theme(panel.grid.major = element_blank()) +
  # theme(panel.grid.minor = element_blank(), panel.background=element_blank(), plot.background=element_blank())

PCoA_1</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-15-2.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="ordination-analysis-by-samples-pca-principal-component-analysis" class="section level3">
<h3>5.2.2. Ordination analysis by samples (PCA, Principal component analysis)</h3>
<pre class="r"><code># make the matrix by reshaping dataframe
library(maditr)
dim(df_ave)</code></pre>
<pre><code>[1] 38075     7</code></pre>
<pre class="r"><code>dc_ave &lt;- dcast(df_ave, SampleID ~ Peptide_id, value.var = &quot;Mean_abundance_ratio&quot;)
dim(dc_ave)</code></pre>
<pre><code>[1]   67 3367</code></pre>
<pre class="r"><code># names(dc_ave)
rownames(dc_ave)</code></pre>
<pre><code> [1] &quot;1&quot;  &quot;2&quot;  &quot;3&quot;  &quot;4&quot;  &quot;5&quot;  &quot;6&quot;  &quot;7&quot;  &quot;8&quot;  &quot;9&quot;  &quot;10&quot; &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot; &quot;15&quot;
[16] &quot;16&quot; &quot;17&quot; &quot;18&quot; &quot;19&quot; &quot;20&quot; &quot;21&quot; &quot;22&quot; &quot;23&quot; &quot;24&quot; &quot;25&quot; &quot;26&quot; &quot;27&quot; &quot;28&quot; &quot;29&quot; &quot;30&quot;
[31] &quot;31&quot; &quot;32&quot; &quot;33&quot; &quot;34&quot; &quot;35&quot; &quot;36&quot; &quot;37&quot; &quot;38&quot; &quot;39&quot; &quot;40&quot; &quot;41&quot; &quot;42&quot; &quot;43&quot; &quot;44&quot; &quot;45&quot;
[46] &quot;46&quot; &quot;47&quot; &quot;48&quot; &quot;49&quot; &quot;50&quot; &quot;51&quot; &quot;52&quot; &quot;53&quot; &quot;54&quot; &quot;55&quot; &quot;56&quot; &quot;57&quot; &quot;58&quot; &quot;59&quot; &quot;60&quot;
[61] &quot;61&quot; &quot;62&quot; &quot;63&quot; &quot;64&quot; &quot;65&quot; &quot;66&quot; &quot;67&quot;</code></pre>
<pre class="r"><code># make Peptide_id more simpler (Pep_1 -&gt; P.1)
names(dc_ave) &lt;- str_replace(names(dc_ave),&quot;Pep_&quot;,&quot;P.&quot;)
# make first column into row name
dc_ave &lt;- tibble::column_to_rownames(dc_ave, var=&quot;SampleID&quot;)

dim(dc_ave) # 67 samples (row) x 3366 peptides (columns)</code></pre>
<pre><code>[1]   67 3366</code></pre>
<pre class="r"><code># First, the typical principal component analysis on the samples would be to transpose the data such that the samples are rows of the data matrix. The prcomp function can be used to return the principal components and other variables.(https://genomicsclass.github.io/book/pages/pca_svd.html)

# pc &lt;- prcomp(dc_ave) # due to NA this is not working

# how should we change NA in the matrix?
any(is.na(dc_ave))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>sum(is.na(dc_ave)) </code></pre>
<pre><code>[1] 187447</code></pre>
<pre class="r"><code>dim(dc_ave)</code></pre>
<pre><code>[1]   67 3366</code></pre>
<pre class="r"><code># let&#39;s replace na into minimum value for each sample
dc_namin &lt;- dc_ave
for (i in 1:dim(dc_ave)[1]){
  rowMin &lt;- (min(dc_ave[i,] ,na.rm=T))
  # print(rowMin)
  for (j in 1:dim(dc_ave)[2]) {
    if (is.na(dc_namin[i,j])){
      dc_namin[i,j] &lt;- rowMin
      # print(dc_namin[i,j])
    }
  }
}

sum(is.na(dc_namin)) # 0 mean success</code></pre>
<pre><code>[1] 0</code></pre>
<pre class="r"><code>dim(dc_namin)</code></pre>
<pre><code>[1]   67 3366</code></pre>
<pre class="r"><code># PCA with prcomp
pc &lt;- prcomp(dc_namin)
eigs &lt;- pc$sdev^2
# variance explained
eigs[1] / sum(eigs) # pc1</code></pre>
<pre><code>[1] 0.5517225</code></pre>
<pre class="r"><code>eigs[2] / sum(eigs) # pc2</code></pre>
<pre><code>[1] 0.071943</code></pre>
<pre class="r"><code># First, a new dataframe should be created, with the information of sample-group. (http://huboqiang.cn/2016/03/03/RscatterPlotPCA)
df_out &lt;- as.data.frame(pc$x)
df_out$depth &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 1 )
df_out$time &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 2 )
df_out$time &lt;- as.numeric(str_replace(df_out$time, &quot;T&quot;,&quot;&quot;))
df_out$time &lt;- factor(df_out$time,levels=1L:8L)
df_out$rep &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 3 )
df_out$depth &lt;- factor(df_out$depth, levels=depth_vec)

head(df_out)</code></pre>
<pre><code>                 PC1        PC2        PC3         PC4         PC5       PC6
BDCL_T1_A  -76.71475  38.712545 -27.732793   4.4072664 -23.7323323 11.682015
BDCL_T1_B   62.99826   1.204047  -4.939825 -11.1902280   0.7549991  7.175836
BDCL_T1_C -100.59088 -51.162269 -20.905828  11.4868257  -1.7228056  2.423109
BDCL_T2_A  -60.76342  22.892838 -27.442097   6.1933357 -13.0708058  1.761372
BDCL_T2_B   26.89529  10.493096  -4.816001 -34.8524060  -7.1085207  3.445134
BDCL_T3_A  -16.11807  32.355056 -28.377670  -0.2760642 -13.1944667 -9.387300
                 PC7         PC8         PC9       PC10     PC11       PC12
BDCL_T1_A 19.4036847 -10.2319665  12.6919065 -0.2692824 9.291826 -63.559937
BDCL_T1_B  0.2420403   3.1351818   2.6018767  0.9119197 3.628302  -1.028216
BDCL_T1_C  2.7284314   0.8295987  -1.8344061 -1.7273747 2.735096  -3.600830
BDCL_T2_A 12.0688250   9.6177917  -1.9263571  9.7689290 3.337344  -3.090871
BDCL_T2_B  2.4931662   1.3498779 -11.2623719  2.2641917 5.280110   2.311361
BDCL_T3_A 16.4875232   0.2659939   0.3589178 10.4077937 2.198834  -6.003224
                 PC13        PC14       PC15       PC16       PC17        PC18
BDCL_T1_A  27.0923320 -17.7670580  1.0603472 -2.0192534  0.6444233   8.4109131
BDCL_T1_B   0.3865763  -1.6646673 -0.1280267 -0.3987174  2.2734450   0.4091348
BDCL_T1_C  -7.1086484   0.2396971  1.6769204  8.1035109 -3.3205996   0.6978478
BDCL_T2_A -13.7674972  40.3643705 -9.6605577  0.6833896 11.6277878 -11.4082968
BDCL_T2_B  -6.1258558  -8.5562640 -0.2087986 11.3332422  3.9925653   5.1361632
BDCL_T3_A  -4.6780138  18.0642464 -6.3904415 15.4004316 -8.1590427  -5.4570795
                PC19        PC20      PC21       PC22      PC23        PC24
BDCL_T1_A -2.1979499   7.4407885 -4.206411   3.196812  4.062263   0.1628943
BDCL_T1_B  0.2579843   3.0065143 -5.369855  -4.464255 -1.273684  -0.1580775
BDCL_T1_C -1.3192153   0.4619432 -6.222981  -6.158967 -2.845323   3.5891178
BDCL_T2_A 17.7840559  39.2623715  1.241722   2.152685  3.203743  -2.7843962
BDCL_T2_B  7.9123358   4.5875251  7.569311 -21.421360  5.369693 -16.8400000
BDCL_T3_A 33.3375545 -41.7316875 -5.853168  -8.155350 -2.548916   7.2296444
                PC25      PC26      PC27       PC28       PC29      PC30
BDCL_T1_A  0.3448015 -2.907847 -1.144081 -0.8884246  1.2041705  1.007728
BDCL_T1_B -4.1806163 -1.950713 -3.008213 -1.0351117  3.3021755 -1.216859
BDCL_T1_C -0.5912255 17.816330  7.815218 13.7168708 -8.2202534  2.923056
BDCL_T2_A -4.4753008  4.802618  1.334516 -6.0574898  6.6329636 -2.910562
BDCL_T2_B -2.9768172 -2.743682 -3.435009 14.7085317 -7.8830668 28.839979
BDCL_T3_A  1.5377327 -5.960728 -1.143289 -2.7529095  0.5480739 -2.380173
                 PC31       PC32       PC33       PC34         PC35       PC36
BDCL_T1_A   0.4168946 -1.1578695 -1.5176244  1.6757144   0.04470535 1.64165161
BDCL_T1_B   0.2279662  6.4142736 -1.3442282 -2.5755168   0.08119817 0.69188357
BDCL_T1_C -10.8271181  5.7608266 -6.1322010 -9.7247092 -12.49160047 4.69559577
BDCL_T2_A   0.3122458 -0.5104172  0.4771393  4.7201211  -0.65167901 1.36415416
BDCL_T2_B  16.4531819 -0.7929184  9.9948680  3.4698155  -1.72865967 6.93981742
BDCL_T3_A   0.1580326 -4.5693385 -5.4617998  0.8904997   0.76152786 0.03794163
                PC37        PC38       PC39       PC40       PC41       PC42
BDCL_T1_A -0.3311464  -0.1658277 -0.1307561 -0.1222908 -0.2444546 -0.2035859
BDCL_T1_B  0.2252707  -6.3835201  8.1735994 -3.4312580  0.7995135 -0.9535802
BDCL_T1_C  5.2318544 -24.4575358 -4.5402718  7.6770215 -2.9268276  6.8858115
BDCL_T2_A  3.0756120   0.1363781 -0.3704383 -1.0468877 -0.1107092  0.5363007
BDCL_T2_B  2.0288745  -0.3524182 -0.5486141 -4.0357259  1.6618837 -2.5454634
BDCL_T3_A  1.5570204  -0.1409182  0.8908182 -1.1370613  1.8536736  0.4755992
                PC43         PC44         PC45       PC46       PC47
BDCL_T1_A -0.2950259   0.44877458   0.93963810 -0.9056214 -0.4736826
BDCL_T1_B -4.5157632 -12.20401149 -14.63316253 22.3114674  4.4006342
BDCL_T1_C -0.7545820   2.03318605   0.09053739 -0.4093168 -0.1292199
BDCL_T2_A  0.3643269  -0.84872338   0.02840034 -2.1229030  0.5619704
BDCL_T2_B  1.1274834   1.09495283   1.47458982 -1.4887158  0.3894406
BDCL_T3_A  0.1158898   0.00276412  -0.58319728 -0.8037418 -0.1955139
                 PC48       PC49         PC50       PC51       PC52       PC53
BDCL_T1_A -0.57019764  0.4346033  0.267260423  0.5739833 -0.3486191 -0.4381481
BDCL_T1_B  8.93471560 -3.7881188  6.098769566 -1.3507501  4.4058013 -1.2458415
BDCL_T1_C -2.60690699  0.9732083  0.378662692 -0.3387400  0.3524069 -0.1795571
BDCL_T2_A -0.01999058  0.7751429 -0.002924125  0.5417404 -0.3930796 -0.9570990
BDCL_T2_B -0.10206817 -1.2154805 -0.709609738 -0.9681095  0.7312141  0.5221322
BDCL_T3_A  0.99107257  0.4436287  0.196207152  0.7435028  0.0416821 -0.4563652
                 PC54       PC55        PC56         PC57         PC58
BDCL_T1_A  0.19821070 0.01983391  0.14198197 -0.001369853 -0.002600525
BDCL_T1_B -1.47649782 0.17552867  0.87135358 -0.287711737  0.067622846
BDCL_T1_C  0.35950316 0.21749728 -1.02491494  0.017580996  0.187416521
BDCL_T2_A -0.33829104 0.22517608 -0.04916386  0.015480106 -0.080920084
BDCL_T2_B  0.08809501 0.69185401  0.07369075 -0.308232860 -0.029125770
BDCL_T3_A  0.05252989 0.28036645  0.14361790 -0.161238809  0.124765627
                  PC59         PC60        PC61         PC62        PC63
BDCL_T1_A -0.042540094  0.000278926  0.01201766  0.006730109  0.02440547
BDCL_T1_B  0.143903851  0.068279548  0.15353058 -0.058325003  0.07840886
BDCL_T1_C  0.013586377  0.290549991 -0.22658397  0.062798956  0.09114341
BDCL_T2_A -0.009017083  0.086061927 -0.02572804 -0.012567292  0.01224441
BDCL_T2_B -0.008853583 -0.035477730  0.01866598 -0.014742491  0.02715270
BDCL_T3_A -0.008537167 -0.020684888  0.02132631 -0.025250785 -0.01800935
                  PC64         PC65         PC66          PC67 depth time rep
BDCL_T1_A  0.011985176 -0.008616089 -0.005261391 -1.185475e-13  BDCL    1   A
BDCL_T1_B -0.003025716  0.050929851  0.024153696  1.523000e-13  BDCL    1   B
BDCL_T1_C  0.139425215 -0.103571723 -0.028377967  2.439368e-14  BDCL    1   C
BDCL_T2_A  0.014575086 -0.007543213 -0.003374602 -2.384724e-14  BDCL    2   A
BDCL_T2_B -0.026773514  0.036295549 -0.004371055  9.389776e-14  BDCL    2   B
BDCL_T3_A  0.016094838  0.003589964  0.010381887 -5.335142e-14  BDCL    3   A</code></pre>
<pre class="r"><code># plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2,label = time))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, alpha=0.9, aes(col= depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis plot for samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
  theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot PCA with timepoint labels
ggplot(df_out,aes(x=PC1,y=PC2,label = time))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 4.5, alpha=0.9, aes(col= depth))+
  scale_fill_manual(values=cols) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis plot for samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
  theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;))+
  ggrepel::geom_label_repel()</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># plot PCA with timepoint gradient

ggplot(df_out,aes(x=PC1,y=PC2,alpha = time))+
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
  geom_point(size = 5, stroke=1.2, aes(col=depth))+
  # geom_point(size = 5,stroke = 1.3, shape=21, aes(fill=depth))+
  scale_fill_manual(values=cols) +
  scale_alpha_manual(values=seq(0.3,1,length.out = 8)) +
  xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(eigs[1] / sum(eigs),3)*100,&quot;%)&quot;))+
  ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(eigs[2] / sum(eigs),3)*100,&quot;%) \n&quot;)) +
  ggtitle(&quot;Principle component analysis plot for samples \n&quot;) +
  ## adjust positions
  guides(fill = guide_legend(ncol = 4,reverse = T))+
  theme(legend.position=&quot;bottom&quot;) +
  theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
  theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
  theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;))</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # ggrepel::geom_label_repel()


# PCA with sparse pca methods

library(sparsepca)
# Sparse Principal Component Analysis (spca).
spc &lt;- spca(dc_namin, k = 2, alpha = 1e-04, beta = 1e-04, center = TRUE,
            scale = FALSE, max_iter = 1000, tol = 1e-05, verbose = 0)
summary(spc)</code></pre>
<pre><code>                            PC1     PC2
Explained variance     5365.714 685.929
Standard deviations      73.251  26.190
Proportion of variance    0.549   0.070
Cumulative proportion     0.549   0.619</code></pre>
<pre class="r"><code># Randomized Sparse Principal Component Analysis (rspca).
rspc &lt;- rspca(dc_namin, k = 2, alpha = 1e-04, beta = 1e-04, center = TRUE,
              scale = FALSE, max_iter = 1000, tol = 1e-05, verbose = 0)
summary(rspc)</code></pre>
<pre><code>                            PC1     PC2
Explained variance     5364.853 687.739
Standard deviations      73.245  26.225
Proportion of variance    0.549   0.070
Cumulative proportion     0.549   0.619</code></pre>
<pre class="r"><code># Robust Sparse Principal Component Analysis (robspca)
robspc &lt;- robspca(dc_namin, k = 2, alpha = 1e-04, beta = 1e-04, gamma = 100,
                  center = TRUE, scale = FALSE, max_iter = 1000, tol = 1e-05, verbose = 0)
summary(robspc)</code></pre>
<pre><code>                            PC1     PC2
Explained variance     5365.714 685.929
Standard deviations      73.251  26.190
Proportion of variance    0.549   0.070
Cumulative proportion     0.549   0.619</code></pre>
<pre class="r"><code>sparse_pca_plot &lt;- function(spc,type=&quot;Sparse&quot;){
  df_spc &lt;- data.frame(summary(spc))
  # variance explained
  df_spc[3,1] # pc1
  df_spc[3,2] # pc2

  df_out &lt;- as.data.frame(spc$scores)
  colnames(df_out) &lt;- c(&quot;PC1&quot;,&quot;PC2&quot;)
  df_out$depth &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 1 )
  df_out$time &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 2 )
  df_out$rep &lt;- sapply( strsplit(as.character(row.names(df_out)), &quot;_&quot;), &quot;[[&quot;, 3 )
  df_out$depth &lt;- factor(df_out$depth, levels=depth_vec)
  
  head(df_out)
  
  ggplot(df_out,aes(x=PC1,y=PC2,color=depth))+
    theme(plot.title = element_text(size = 20,hjust = 0.5, face=&quot;bold&quot;)) +
    geom_point(size = 4,alpha=0.7)+
    scale_fill_manual(values=cols) +
    xlab(paste0(&#39;\n PC1&#39;, &quot; (explained variance: &quot;,round(df_spc[3,1],3)*100,&quot;%)&quot;))+
    ylab(paste0(&quot;PC2&quot;,&quot; (explained variance: &quot;,round(df_spc[3,2],3)*100,&quot;%) \n&quot;)) +
    ggtitle(paste0(type,&quot; principle component analysis plot for samples \n&quot;)) +
    ## adjust positions
    guides(fill = guide_legend(ncol = 4,reverse = T))+
    theme(legend.position=&quot;bottom&quot;) +
    theme(plot.title = element_text(size = 20,hjust = 0.5, face=&#39;bold&#39;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, face=&#39;bold&#39;)) + 
    theme(axis.text.x = element_text(angle = 90, hjust = 0.95,face=&#39;bold&#39;, vjust=0.4,size=15))+
    theme(axis.text.y = element_text(size=15, face=&#39;bold&#39;))
    # theme(panel.grid.major = element_blank()) +
    # theme(panel.grid.minor = element_blank(), panel.background=element_blank(), plot.background=element_blank())
}

sparse_pca_plot(spc,type=&quot;Sparse&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sparse_pca_plot(rspc,type=&quot;Randomized Sparse&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sparse_pca_plot(robspc,type=&quot;Robust Sparse&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-16-6.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="rain-analysis-rhythmicity-analysis-incorporating-non-parametric-methods" class="section level3">
<h3>6. RAIN analysis (Rhythmicity Analysis Incorporating Non-parametric methods)</h3>
</div>
<div id="checking-the-abundance_ratio-distribution-of-peptides-in-each-sample" class="section level3">
<h3>6.1. Checking the abundance_ratio distribution of peptides in each sample</h3>
<ol style="list-style-type: decimal">
<li>Peptide level histograms</li>
</ol>
<pre class="r"><code># function to plot histogram
hist_abundance_ratio &lt;- function(depth=&quot;SRF&quot;,time=&quot;T1&quot;){
  df_ave_SRF_T1 &lt;- df_ave %&gt;% filter(Time==time &amp; Depth ==depth)
  theme_set(theme_bw())
  h1 &lt;- ggplot(df_ave_SRF_T1, aes(x=Mean_abundance_ratio)) +
    geom_freqpoly(binwidth = 0.05, aes(color=Replicate)) +
    scale_fill_brewer(palette=&#39; Set2&#39;) +
    ylab(&quot;Mean protein abundance ratio (log2(16O/18O)) \n&quot;) +
    xlab(&quot;\n Frequency of ratio&quot;) +
    ggtitle(paste0(&quot;Histogram of sample (&quot;,depth,&quot;-&quot;,time,&quot;) \n&quot;))+
    ## adjust positions
    theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
    theme(axis.text.y = element_text(size=10))
    # theme(panel.grid.major = element_blank())
    # theme(panel.grid.minor = element_blank()) 
  return(h1)
}

hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T1&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T2&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T3&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T4&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T5&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T6&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-6.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T7&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-7.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio(depth=&quot;SRF&quot;,time=&quot;T8&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-8.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># draw histogram in multigrids
library(gridExtra)
paste0(&quot;T&quot;,1L:8L)</code></pre>
<pre><code>[1] &quot;T1&quot; &quot;T2&quot; &quot;T3&quot; &quot;T4&quot; &quot;T5&quot; &quot;T6&quot; &quot;T7&quot; &quot;T8&quot;</code></pre>
<pre class="r"><code>l &lt;- list()
for (i in paste0(&quot;T&quot;,1L:8L)){
  print(i)
  l[[i]] &lt;- hist_abundance_ratio(depth=&quot;SRF&quot;,time=i)
}</code></pre>
<pre><code>[1] &quot;T1&quot;
[1] &quot;T2&quot;
[1] &quot;T3&quot;
[1] &quot;T4&quot;
[1] &quot;T5&quot;
[1] &quot;T6&quot;
[1] &quot;T7&quot;
[1] &quot;T8&quot;</code></pre>
<pre class="r"><code>length(l)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>grid.arrange(grobs = l, ncol=3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-17-9.png" width="1056" style="display: block; margin: auto;" /></p>
<ol start="2" style="list-style-type: decimal">
<li>Spectra level histograms</li>
</ol>
<pre class="r"><code># function to plot histogram
hist_abundance_ratio_spec &lt;- function(depth=&quot;SRF&quot;,time=&quot;T1&quot;){
  df_int_SRF_T1 &lt;- df_int %&gt;% filter(Time==time &amp; Depth ==depth) %&gt;% select(-Peptide_id)
  theme_set(theme_bw())
  h1 &lt;- ggplot(df_int_SRF_T1, aes(x=cent.log2.16O.18O.Ratio)) +
    geom_freqpoly(binwidth = 0.05, aes(color=Replicate)) +
    scale_fill_brewer(palette=&#39;Set2&#39;) +
    ylab(&quot;Centered log2(16O/18O) ratio \n&quot;) +
    xlab(&quot;\n Frequency of ratio (Spec-level)&quot;) +
    ggtitle(paste0(&quot;Histogram of sample (&quot;,depth,&quot;-&quot;,time,&quot;) \n&quot;))+
    ## adjust positions
    theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5, vjust=0.3,size=13, family=&quot;serif&quot;))+
    theme(axis.text.y = element_text(size=10))
    # theme(panel.grid.major = element_blank())
    # theme(panel.grid.minor = element_blank()) 
  return(h1)
}

hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T1&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T2&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T3&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T4&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T5&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T6&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-6.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T7&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-7.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=&quot;T8&quot;)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-8.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># draw histogram in multigrids
library(gridExtra)
paste0(&quot;T&quot;,1L:8L)</code></pre>
<pre><code>[1] &quot;T1&quot; &quot;T2&quot; &quot;T3&quot; &quot;T4&quot; &quot;T5&quot; &quot;T6&quot; &quot;T7&quot; &quot;T8&quot;</code></pre>
<pre class="r"><code>l &lt;- list()
for (i in paste0(&quot;T&quot;,1L:8L)){
  print(i)
  l[[i]] &lt;- hist_abundance_ratio_spec(depth=&quot;SRF&quot;,time=i)
}</code></pre>
<pre><code>[1] &quot;T1&quot;
[1] &quot;T2&quot;
[1] &quot;T3&quot;
[1] &quot;T4&quot;
[1] &quot;T5&quot;
[1] &quot;T6&quot;
[1] &quot;T7&quot;
[1] &quot;T8&quot;</code></pre>
<pre class="r"><code>length(l)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>grid.arrange(grobs = l, ncol=3)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-18-9.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
<div id="applying-rain-analysis" class="section level3">
<h3>6.2. Applying rain analysis</h3>
<pre class="r"><code># setting our data into input format
df_8_S &lt;- df_filt8 %&gt;% filter(Depth==&#39;SRF&#39;) %&gt;% select(Peptide_id,Time,Ave_abundance_ratio)

# reshape the dataframe with timepoints
library(maditr)
v_names &lt;- c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;)
df_8$Time &lt;- factor(df_8$Time, levels=v_names)
dc_8 &lt;- dcast(df_8, Peptide_id ~ Time, value.var = &quot;Ave_abundance_ratio&quot;)
# make Peptide_id more simpler (Pep_1 -&gt; 1)
dc_8$Peptide_id &lt;- str_replace(dc_8$Peptide_id,&quot;Pep_&quot;,&quot;P.&quot;)
# make first column into row name
dc_8 &lt;- tibble::column_to_rownames(dc_8, var=&quot;Peptide_id&quot;)

# t(dc_8) %&gt;% head
# setup time vector for RAIN (source: Maria&#39;s code provided to me 3/3/21)
# The following code gives a vector where the times given in &quot;t&quot; are represented by a 1 in the vector &quot;ft&quot;.
# The time series in question lasted 84 hours and sampling times were every 4 hours, so the 
# vector &quot;ft&quot; gives a 1 for each hour representing a sampling time, and ft could is a vector of 12 1&#39;s and deltat=4 in the RAIN call below. 
t &lt;- c(0, 3, 6, 9, 15, 19, 22, 25)
(t &lt;- t+1) # start from 1</code></pre>
<pre><code>[1]  1  4  7 10 16 20 23 26</code></pre>
<pre class="r"><code># If the time series is not equally spaced it can be regularized by introducing time points with
# zero repeats. This may be combined with any of the above repeat settings.
# Example: A series measured at times {1, 3, 4, 6} is usable with the measure.sequence
# {1, 0, 1, 1, 0, 1} and a matrix x
t &lt;- c(1,0,0,1,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,1,0,0,1,0,0,1) # adjusted for irregular repeats

# RAIN analysis
library(rain)
rainresult &lt;- rain(t(dc_8), period=8, measure.sequence=t, deltat=1, method=&#39;independent&#39;)

# According to the original paper, adjusting p-values for multiple testing is not necessary (already done in the output): https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4266694/
# rainresult$fdr &lt;- p.adjust(as.numeric(rainresult$pVal), method=&quot;fdr&quot;) 

# pVal cut off 0.25
dim(rainresult)</code></pre>
<pre><code>[1] 532   4</code></pre>
<pre class="r"><code>rainresult %&gt;% filter(pVal &lt; 0.25)</code></pre>
<pre><code>             pVal phase peak.shape period
P.102 0.037634409     1          3      8
P.113 0.118811881     3          5      8
P.114 0.202349583     7          4      8
P.117 0.172174840     2          5      8
P.123 0.020815812     7          3      8
P.130 0.188603869     2          6      8
P.149 0.206451613     2          6      8
P.15  0.198924731     8          6      8
P.152 0.047939445     2          6      8
P.161 0.191699605     4          4      8
P.162 0.159587889     1          6      8
P.171 0.193439865     1          6      8
P.174 0.005444327     7          3      8
P.19  0.002342263     4          3      8
P.190 0.201713028     8          2      8
P.192 0.114922207     1          6      8
P.215 0.224474348     2          4      8
P.229 0.005971198     2          5      8
P.256 0.118811881     8          3      8
P.257 0.239855072     3          4      8
P.26  0.055081734     7          3      8
P.262 0.214521452     7          2      8
P.265 0.220021075     2          4      8
P.271 0.047284439     6          6      8
P.275 0.079569892     3          6      8
P.28  0.214521452     1          3      8
P.284 0.174095879     1          6      8
P.29  0.173181710     7          4      8
P.290 0.202349583     7          4      8
P.296 0.108910891     8          3      8
P.300 0.047284439     4          3      8
P.304 0.062189055     7          3      8
P.307 0.074257426     4          4      8
P.316 0.180403701     8          6      8
P.317 0.087047939     3          5      8
P.33  0.017590618     5          5      8
P.345 0.054455446     8          4      8
P.364 0.047939445     8          2      8
P.398 0.005619951     2          5      8
P.4   0.037634409     8          6      8
P.40  0.209677419     5          3      8
P.411 0.125603865     6          6      8
P.416 0.075436115     2          4      8
P.421 0.155844156     7          3      8
P.442 0.151279318     3          4      8
P.444 0.118811881     3          5      8
P.446 0.055081734     7          3      8
P.464 0.078500707     4          4      8
P.47  0.118279570     4          4      8
P.475 0.193439865     1          6      8
P.482 0.134115139     2          5      8
P.49  0.122383253     6          6      8
P.492 0.206859972     6          6      8
P.538 0.155844156     2          5      8
P.554 0.214521452     1          6      8
P.58  0.186374616     8          4      8
P.593 0.156765677     4          5      8
P.596 0.156765677     1          3      8
P.597 0.012082445     7          3      8
P.644 0.120129870     5          6      8
P.679 0.047939445     2          6      8
P.68  0.235607676     7          3      8
P.682 0.005444327     7          3      8
P.69  0.144839701     4          4      8
P.711 0.174257426     2          4      8
P.725 0.079793944     8          3      8
P.848 0.174095879     8          2      8
P.851 0.155844156     2          5      8
P.87  0.134115139     7          3      8</code></pre>
<pre class="r"><code>rainresult %&gt;% filter(pVal &lt; 0.25) %&gt;% dim() # 69 peptides</code></pre>
<pre><code>[1] 69  4</code></pre>
<pre class="r"><code># pVal cut off 0.15
rainresult %&gt;% filter(pVal &lt; 0.15)</code></pre>
<pre><code>             pVal phase peak.shape period
P.102 0.037634409     1          3      8
P.113 0.118811881     3          5      8
P.123 0.020815812     7          3      8
P.152 0.047939445     2          6      8
P.174 0.005444327     7          3      8
P.19  0.002342263     4          3      8
P.192 0.114922207     1          6      8
P.229 0.005971198     2          5      8
P.256 0.118811881     8          3      8
P.26  0.055081734     7          3      8
P.271 0.047284439     6          6      8
P.275 0.079569892     3          6      8
P.296 0.108910891     8          3      8
P.300 0.047284439     4          3      8
P.304 0.062189055     7          3      8
P.307 0.074257426     4          4      8
P.317 0.087047939     3          5      8
P.33  0.017590618     5          5      8
P.345 0.054455446     8          4      8
P.364 0.047939445     8          2      8
P.398 0.005619951     2          5      8
P.4   0.037634409     8          6      8
P.411 0.125603865     6          6      8
P.416 0.075436115     2          4      8
P.444 0.118811881     3          5      8
P.446 0.055081734     7          3      8
P.464 0.078500707     4          4      8
P.47  0.118279570     4          4      8
P.482 0.134115139     2          5      8
P.49  0.122383253     6          6      8
P.597 0.012082445     7          3      8
P.644 0.120129870     5          6      8
P.679 0.047939445     2          6      8
P.682 0.005444327     7          3      8
P.69  0.144839701     4          4      8
P.725 0.079793944     8          3      8
P.87  0.134115139     7          3      8</code></pre>
<pre class="r"><code>rainresult %&gt;% filter(pVal &lt; 0.15) %&gt;% dim() # 37 peptides</code></pre>
<pre><code>[1] 37  4</code></pre>
<pre class="r"><code># rain analysis function
rain_analysis &lt;- function(df_filt8,depth=&quot;SRF&quot;){
  # setting our data into input format
  df_8 &lt;- df_filt8 %&gt;% filter(Depth==depth) %&gt;% select(Peptide_id,Time,Ave_abundance_ratio)
  # reshape the dataframe with timepoints
  library(maditr)
  v_names &lt;- c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;)
  df_8$Time &lt;- factor(df_8$Time, levels=v_names)
  dc_8 &lt;- dcast(df_8, Peptide_id ~ Time, value.var = &quot;Ave_abundance_ratio&quot;)
  # make Peptide_id more simpler (Pep_1 -&gt; 1)
  dc_8$Peptide_id &lt;- str_replace(dc_8$Peptide_id,&quot;Pep_&quot;,&quot;P.&quot;)
  # make first column into row name
  dc_8 &lt;- tibble::column_to_rownames(dc_8, var=&quot;Peptide_id&quot;)
  t &lt;- c(1,0,0,1,0,0,1,0,0,1,0,0,0,0,0,1,0,0,0,1,0,0,1,0,0,1) # adjusted for irregular repeats
  # RAIN analysis
  library(rain)
  rainresult &lt;- rain(t(dc_8), period=8, measure.sequence=t, deltat=1, method=&#39;independent&#39;)
  return(rainresult)
}</code></pre>
</div>
<div id="rain---p_value-and-peak-timepoint-surface-dcl-bdcl" class="section level3">
<h3>6.3. RAIN - p_value and peak timepoint: Surface / DCL / BDCL</h3>
<pre class="r"><code># function to plot p-value
time_series_pvalue &lt;- function(df_filt8,depth=&quot;SRF&quot;,threshold=0.25){
  rain_SRF8 &lt;-rain_analysis(df_filt8,depth=depth)
  print(sum(rain_SRF8$pVal&lt;=threshold))
  plot((rain_SRF8$pVal) ~ rain_SRF8$phase, xlab=&quot;Peak timepoint (Sampling order #)&quot;, ylab=&quot;p-value&quot;,main=paste0(&quot;Rhythmic peptides (&quot;,depth,&quot;)&quot;),cex=2,cex.lab=2, cex.main=2,cex.axis=1.6)
  points(rain_SRF8$pVal[rain_SRF8$pVal&lt;=threshold] ~ rain_SRF8$phase[rain_SRF8$pVal&lt;=threshold], col=&quot;red&quot;,cex=2)
  abline(h=threshold, col=&quot;red&quot;)
}

time_series_pvalue(df_filt8,depth=&quot;SRF&quot;,threshold=0.25)</code></pre>
<pre><code>[1] 69</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_pvalue(df_filt8,depth=&quot;DCL&quot;,threshold=0.25)</code></pre>
<pre><code>[1] 52</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>time_series_pvalue(df_filt8,depth=&quot;BDCL&quot;,threshold=0.25)</code></pre>
<pre><code>[1] 56</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># draw plots in multigrids
par(mfrow = c(1, 3))
# filtered for 8 timepoint appearing peptides
for (i in 1:length(depth_vec)){
  time_series_pvalue(df_filt8,depth=depth_vec[i],threshold=0.25)
}</code></pre>
<pre><code>[1] 69</code></pre>
<pre><code>[1] 52</code></pre>
<pre><code>[1] 56</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-4.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># filtered for 7, 8 timepoint appearing peptides
for (i in 1:length(depth_vec)){
  time_series_pvalue(df_filt78,depth=depth_vec[i],threshold=0.25)
}</code></pre>
<pre><code>[1] 70</code></pre>
<pre><code>[1] 52</code></pre>
<pre><code>[1] 56</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-5.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># change the threshold to 0.15
for (i in 1:length(depth_vec)){
  time_series_pvalue(df_filt8,depth=depth_vec[i],threshold=0.15)
}</code></pre>
<pre><code>[1] 37</code></pre>
<pre><code>[1] 32</code></pre>
<pre><code>[1] 30</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-6.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>for (i in 1:length(depth_vec)){
  time_series_pvalue(df_filt78,depth=depth_vec[i],threshold=0.15)
}</code></pre>
<pre><code>[1] 38</code></pre>
<pre><code>[1] 32</code></pre>
<pre><code>[1] 30</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-20-7.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>null device 
          1 </code></pre>
</div>
<div id="rhythmic-peptides-peak-time-distribution" class="section level3">
<h3>6.4. Rhythmic peptide’s peak time distribution</h3>
<pre class="r"><code>rain_SRF8 &lt;- rain_analysis(df_filt8,depth=&quot;SRF&quot;) %&gt;% mutate(Depth=&quot;SRF&quot;) %&gt;% tibble::rownames_to_column(var = &quot;Peptide_id&quot;)
rain_DCL8 &lt;- rain_analysis(df_filt8,depth=&quot;DCL&quot;) %&gt;% mutate(Depth=&quot;DCL&quot;) %&gt;% tibble::rownames_to_column(var = &quot;Peptide_id&quot;)
rain_BDCL8 &lt;- rain_analysis(df_filt8,depth=&quot;BDCL&quot;) %&gt;% mutate(Depth=&quot;BDCL&quot;) %&gt;% tibble::rownames_to_column(var = &quot;Peptide_id&quot;)
dim(rain_SRF8)</code></pre>
<pre><code>[1] 532   6</code></pre>
<pre class="r"><code>dim(rain_DCL8)</code></pre>
<pre><code>[1] 426   6</code></pre>
<pre class="r"><code>dim(rain_BDCL8)</code></pre>
<pre><code>[1] 392   6</code></pre>
<pre class="r"><code>rain_8 &lt;- rbind(rain_SRF8, rain_DCL8) %&gt;% rbind(rain_BDCL8)
dim(rain_8)</code></pre>
<pre><code>[1] 1350    6</code></pre>
<pre class="r"><code>tail(rain_8)</code></pre>
<pre><code>     Peptide_id      pVal phase peak.shape period Depth
1345       P.94 0.5550883     2          6      8  BDCL
1346       P.95 0.3677419     8          3      8  BDCL
1347       P.96 0.9813643     5          4      8  BDCL
1348       P.97 0.8602151     3          5      8  BDCL
1349       P.98 0.8762820     2          6      8  BDCL
1350       P.99 0.6112296     7          3      8  BDCL</code></pre>
<pre class="r"><code>rain_sig &lt;- rain_8 %&gt;% filter(pVal &lt;= 0.25)
rain_sig$Depth &lt;- factor(rain_sig$Depth, levels=depth_vec)

ggplot(rain_sig, aes(x=phase)) +
  geom_freqpoly(binwidth = 1, aes(color=Depth), size=1.5) +
  # geom_histogram(binwidth = 1, position=&quot;dodge&quot;, aes(fill=Depth)) +
  scale_color_manual(values=cols) +
  ylab(&quot;Rhythmic peptide frequency \n&quot;) +
  # xlab(&quot;\n Peak timepoint (Sampling order #)&quot;) +
  xlab(&quot;\n Time of day on June 26-27 2019 (Daylight 05:17-20:30)&quot;) +
  scale_x_continuous(breaks=seq(1,8,1),labels =c(df_time$date_time))+
  ggtitle(paste0(&quot;Histogram of peak time \n&quot;))+
  # scale_x_continuous(breaks=seq(1,8,1)) +
  ## adjust positions
  theme(plot.title = element_text(size = 20,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.x = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.title.y = element_text(size = 15,hjust = 0.5, family=&quot;serif&quot;)) + 
  theme(axis.text.x = element_text(hjust = 0.9, vjust=0.9,angle=40,size=13, family=&quot;serif&quot;))+
  theme(axis.text.y = element_text(size=10)) +
  annotate(&quot;rect&quot;, xmin = 3.6, xmax = 5.35, ymin = -Inf, ymax = Inf, alpha = .2) +
  theme(panel.grid.minor = element_blank())</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-21-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  # theme(panel.grid.major = element_blank())</code></pre>
</div>
<div id="amplitude" class="section level3">
<h3>6.5. Amplitude</h3>
<pre class="r"><code># Shaping data
rain_sig1 &lt;- rain_8 %&gt;% filter(pVal &lt;= 0.25)
rain_sig1$Peptide_id &lt;- str_replace(rain_sig1$Peptide_id,&quot;P.&quot;,&quot;Pep_&quot;)
rain_sig1$phase &lt;- str_replace(rain_sig1$phase,&quot;^&quot;,&quot;T&quot;)
rain_sig2 &lt;- rain_sig1 %&gt;% rename(Time = phase) %&gt;% select(-period)

# rain_sig2 %&gt;% arrange(Peptide_id)
           
# plot
time_series_1depth_1peak &lt;- function(df_filt8,n=1,time=&quot;T1&quot;){
  peak_T1 &lt;- rain_sig2 %&gt;% filter(Time==time) %&gt;% select(Peptide_id)
  print(length(peak_T1$Peptide_id))
  time_series_clust_wError(df_filt8, n, peak_T1$Peptide_id) +
    ggtitle(paste0(&quot;Peptides with peak time &quot;,time))+
    theme(axis.title.x = element_text(size = 10,hjust = 0.5, family=&quot;serif&quot;)) + 
    theme(axis.title.y = element_text(size = 10,hjust = 0.5, family=&quot;serif&quot;))
}


# draw histogram in multigrids
library(gridExtra)

# (1) Surface
l &lt;- list()
for (i in paste0(&quot;T&quot;,1L:8L)){
  print(i)
  l[[i]] &lt;- time_series_1depth_1peak(df_filt8,n=1,time=i)
}</code></pre>
<pre><code>[1] &quot;T1&quot;
[1] 23
[1] &quot;T2&quot;
[1] 40
[1] &quot;T3&quot;
[1] 16
[1] &quot;T4&quot;
[1] 22
[1] &quot;T5&quot;
[1] 9
[1] &quot;T6&quot;
[1] 5
[1] &quot;T7&quot;
[1] 33
[1] &quot;T8&quot;
[1] 29</code></pre>
<pre class="r"><code>length(l)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>grid.arrange(grobs = l, ncol=4)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-22-1.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (2) DCL
l &lt;- list()
for (i in paste0(&quot;T&quot;,1L:8L)){
  print(i)
  l[[i]] &lt;- time_series_1depth_1peak(df_filt8,n=2,time=i)
}</code></pre>
<pre><code>[1] &quot;T1&quot;
[1] 23
[1] &quot;T2&quot;
[1] 40
[1] &quot;T3&quot;
[1] 16
[1] &quot;T4&quot;
[1] 22
[1] &quot;T5&quot;
[1] 9
[1] &quot;T6&quot;
[1] 5
[1] &quot;T7&quot;
[1] 33
[1] &quot;T8&quot;
[1] 29</code></pre>
<pre class="r"><code>length(l)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>grid.arrange(grobs = l, ncol=4)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-22-2.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># (3) BDCL
l &lt;- list()
for (i in paste0(&quot;T&quot;,1L:8L)){
  print(i)
  l[[i]] &lt;- time_series_1depth_1peak(df_filt8,n=3,time=i)
}</code></pre>
<pre><code>[1] &quot;T1&quot;
[1] 23
[1] &quot;T2&quot;
[1] 40
[1] &quot;T3&quot;
[1] 16
[1] &quot;T4&quot;
[1] 22
[1] &quot;T5&quot;
[1] 9
[1] &quot;T6&quot;
[1] 5
[1] &quot;T7&quot;
[1] 33
[1] &quot;T8&quot;
[1] 29</code></pre>
<pre class="r"><code>length(l)</code></pre>
<pre><code>[1] 8</code></pre>
<pre class="r"><code>grid.arrange(grobs = l, ncol=4)</code></pre>
<p><img src="figure/peptide_level_visualization.Rmd/unnamed-chunk-22-3.png" width="1056" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Getting amplitude
inner_join(rain_sig2, df_ave_rep, by=c(&quot;Peptide_id&quot;=&quot;Peptide_id&quot;,&quot;Depth&quot;=&quot;Depth&quot;,&quot;Time&quot;=&quot;Time&quot;)) %&gt;% dim()</code></pre>
<pre><code>[1] 177   7</code></pre>
</div>
<div id="archived-not-using-this-part" class="section level3">
<h3>Archived: Not using this part</h3>
<p>Approach2: Time Series Classification using Dynamic time warping distance Time series classification is to build a classification model based on labelled time series and then use the model to predict the label of unlabelled time series. The way for time series classification with R is to extract and build features from time series data first, and then apply existing classification techniques, such as SVM, k-NN, neural networks, regression and decision trees, to the feature set.</p>
<p>Discrete Wavelet Transform (DWT) provides a multi-resolution representation using wavelets and is used in the example below. Another popular feature extraction technique is Discrete Fourier Transform (DFT). (<a href="http://www.rdatamining.com/examples/time-series-clustering-classification" class="uri">http://www.rdatamining.com/examples/time-series-clustering-classification</a>)</p>
<pre class="r"><code># extracting DWT coefficients (with Haar filter)
library(wavelets)
wtData &lt;- NULL

sc &lt;- read.table(&quot;data/synthetic_control.data.txt&quot;, header=F, sep=&quot;&quot;)

for (i in 1:nrow(sc)){
  a &lt;- t(sc[i,])
  wt &lt;- dwt(a, filter=&quot;haar&quot;, boundary=&#39;periodic&#39;)
  wtData &lt;- rbind(wtData, unlist(c(wt@W,wt@V[[wt@level]])))
}

wtData &lt;- as.data.frame(wtData)
head(wtData)

# set class labels into categorical values
classId &lt;- c(rep(1,100), rep(2,100), rep(3,100),rep(4,100), rep(5,100), rep(6,100))
wtSc &lt;- data.frame(cbind(classId, wtData))

# build a decision tree with ctree() in package party
library(party)
ct &lt;- ctree(classId ~ ., data=wtSc, controls = ctree_control(minsplit=30, minbucket=10, maxdepth=5))
pClassId &lt;- predict(ct)

# check predicted classes against original class labels
table(classId, pClassId)
# accuracy
(sum(classId==pClassId)) / nrow(wtSc)

plot(ct, ip_args=list(pval=FALSE), ep_args=list(digits=0))


# Lower CVI means better clustering : k_shape is the best clustering method
sapply(list(DTW = pc_dtw, DTW_LB = pc_dtwlb, kShape = pc_ks, TADPole = pc_tp), cvi, b = names(l_SRF8), type = &quot;VI&quot;)

sapply(list(DTW = pc_dtw, DTW_LB = pc_dtwlb, kShape = pc_ks, TADPole = pc_tp), cvi, type = &quot;internal&quot;)


# subset
data &lt;- CharTraj[1L:20L]
pc_k &lt;- tsclust(data, k = 3L:5L, distance = &quot;dtw_basic&quot;, centroid = &quot;pam&quot;, seed = 94L)
names(pc_k) &lt;- paste0(&quot;k_&quot;, 3L:5L)
sapply(pc_k, cvi, type = &quot;internal&quot;)



names(l_SRF8)</code></pre>
<p>5.3. dtwclust the clustering itself may be shape-based, feature-based or model-based.<br />
In the context of shape-based time-series clustering, it is common to utilize the Dynamic Time Warping (DTW) distance as dissimilarity measure (Aghabozorgi et al. 2015).<br />
DTW is a dynamic programming algorithm that compares two series and tries to find the optimum warping path between them under certain constraints, such as monotonicity. It started being used by the data mining community to overcome some of the limitations associated with the Euclidean distance (Ratanamahatana and Keogh 2004; Berndt and Clifford 1994).<br />
<a href="https://cran.r-project.org/web/packages/dtwclust/vignettes/dtwclust.pdf" class="uri">https://cran.r-project.org/web/packages/dtwclust/vignettes/dtwclust.pdf</a></p>
<pre class="r"><code># install.packages(&#39;dtwclust&#39;)
# install.packages(&#39;TSclust&#39;)
library(dtwclust)
require(&quot;TSclust&quot;)
data(&quot;uciCT&quot;)

proxy::pr_DB$set_entry(FUN = diss.ACF, names = c(&quot;ACFD&quot;), loop = TRUE, type = &quot;metric&quot;, distance = TRUE, description = &quot;Autocorrelation-based distance&quot;)

# Normalized DTW
ndtw &lt;- function(x, y, ...) {
  dtw(x, y, ...,
  step.pattern = asymmetric,
  distance.only = TRUE)$normalizedDistance
}

# Register the distance with proxy
proxy::pr_DB$set_entry(FUN = ndtw, names = c(&quot;nDTW&quot;),
  loop = TRUE, type = &quot;metric&quot;, distance = TRUE,
  description = &quot;Normalized, asymmetric DTW&quot;)

# Partitional clustering
tsclust(CharTraj[1L:10L], k = 2L,
  distance = &quot;nDTW&quot;, seed = 838)

# hierarchical clustering
hc_sbd &lt;- tsclust(CharTraj, type = &quot;h&quot;, k = 20L,
  preproc = zscore, seed = 899,
  distance = &quot;sbd&quot;, centroid = shape_extraction,
  control = hierarchical_control(method = &quot;average&quot;))

plot(hc_sbd)

# The series and the obtained prototypes can be plotted too
plot(hc_sbd, type = &quot;sc&quot;)

# Focusing on the first cluster
plot(hc_sbd, type = &quot;series&quot;, clus = 1L)
plot(hc_sbd, type = &quot;centroids&quot;, clus = 1L)


### Hierarchical clustering w/ shape based distance
hc_sbd &lt;- tsclust(CharTraj, type = &quot;h&quot;, k = 20L,
  preproc = zscore, seed = 899,
  distance = &quot;sbd&quot;, centroid = shape_extraction,
  control = hierarchical_control(method = &quot;average&quot;))


### partial clustering methods
# Reinterpolate to same length
data &lt;- reinterpolate(CharTraj, new.length = max(lengths(CharTraj)))

# z-normalization
data &lt;- zscore(data[60L:100L])
names(data)
# DTW method
pc_dtw &lt;- tsclust(data, k = 4L,
  distance = &quot;dtw_basic&quot;, centroid = &quot;dba&quot;,
  trace = TRUE, seed = 8,
  norm = &quot;L2&quot;, window.size = 20L,
  args = tsclust_args(cent = list(trace = TRUE)))

# DTW_lb method
pc_dtwlb &lt;- tsclust(data, k = 4L,
  distance = &quot;dtw_lb&quot;, centroid = &quot;dba&quot;,
  trace = TRUE, seed = 8,
  norm = &quot;L2&quot;, window.size = 20L,
  control = partitional_control(pam.precompute = FALSE),
  args = tsclust_args(cent = list(trace = TRUE)))

# k-shape method
pc_ks &lt;- tsclust(data, k = 4L,
  distance = &quot;sbd&quot;, centroid = &quot;shape&quot;,
  seed = 8, trace = TRUE)

# Tadpole method
pc_tp &lt;- tsclust(data, k = 4L, type = &quot;t&quot;,
  seed = 8, trace = TRUE,
  control = tadpole_control(dc = 1.5,
  window.size = 20L))

# Fuzzy clustering

# Calculate autocorrelation up to 50th lag
acf_fun &lt;- function(series, ...) {
  lapply(series, function(x) {
  as.numeric(acf(x, lag.max = 50, plot = FALSE)$acf)
  })
}
# Fuzzy c-means
fc &lt;- tsclust(CharTraj[1:20], type = &quot;f&quot;, k = 4L,
  preproc = acf_fun, distance = &quot;L2&quot;,
  seed = 42)

# Lower CVI means better clustering : k_shape is the best clustering method
sapply(list(DTW = pc_dtw, DTW_LB = pc_dtwlb, kShape = pc_ks, TADPole = pc_tp), cvi, b = CharTrajLabels[60L:100L], type = &quot;VI&quot;)

CharTraj

# Plot crisp partition in the original space
plot(fc, series = CharTraj[1:20], type = &quot;series&quot;)

# subset
data &lt;- CharTraj[1L:20L]
pc_k &lt;- tsclust(data, k = 3L:5L, distance = &quot;dtw_basic&quot;, centroid = &quot;pam&quot;, seed = 94L)
names(pc_k) &lt;- paste0(&quot;k_&quot;, 3L:5L)
sapply(pc_k, cvi, type = &quot;internal&quot;)</code></pre>
<p>6.1. Given code: <a href="https://rdrr.io/bioc/rain/f/inst/doc/rain.pdf" class="uri">https://rdrr.io/bioc/rain/f/inst/doc/rain.pdf</a></p>
<pre class="r"><code># BiocManager::install(&#39;rain&#39;)

set.seed(123)
times &lt;- c(1: 24) * 2
sin &lt;- 1 + 0.5 * sin(times / 24 * 2 * pi) + rnorm(24, 0, 0.3)
saw &lt;- rep(13:24 / 18 , 2) + rnorm(24, 0, 0.3)
measure &lt;- cbind(sin, saw)
require(&#39;lattice&#39;)
xyplot(t(measure)~rep(times, each=2) | c(&#39;sin&#39;, &#39;saw&#39;),layout = c(1, 2), type = &#39;o&#39;, xlab = &#39;time&#39;, ylab = &#39;value&#39;, cex.lab = 0.6)

require(rain)
rainresult &lt;- rain(measure, period=24, deltat=2, peak.border=c(0.1,0.9), verbose=FALSE)
rainresult

data(menetRNASeqMouseLiver)
colnames(menetRNASeqMouseLiver)

results &lt;- rain(t(menetRNASeqMouseLiver), deltat=4, period=24, nr.series=2, peak.border=c(0.3, 0.7), verbose=FALSE)
best &lt;- order(results$pVal)[1:10]
xyplot(as.matrix(menetRNASeqMouseLiver[best, (0:5 * 2 + rep(c(1, 2), each = 6))]) ~rep(0:11 * 4 + 2, each = 10) |rownames(menetRNASeqMouseLiver)[best], scales = list(y = list(relation = &#39;free&#39;)), layout = c(2, 5), type = &#39;b&#39;, pch = 16, xlab = &#39;time&#39;, ylab = &#39;expression value&#39;, cex.lab = 1)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] rain_1.24.0         multtest_2.46.0     Biobase_2.50.0     
 [4] BiocGenerics_0.36.0 gmp_0.6-2           gridExtra_2.3      
 [7] sparsepca_0.1.2     ggrepel_0.9.1       labdsv_2.0-1       
[10] mgcv_1.8-33         nlme_3.1-149        dtw_1.22-3         
[13] proxy_0.4-24        maditr_0.7.4        data.table_1.14.0  
[16] formattable_0.2.1   ggplot2_3.3.3       stringr_1.4.0      
[19] tidyr_1.1.2         dplyr_1.0.4         workflowr_1.6.2    

loaded via a namespace (and not attached):
 [1] sass_0.3.1        jsonlite_1.7.2    splines_4.0.3     bslib_0.2.4      
 [5] assertthat_0.2.1  highr_0.8         stats4_4.0.3      yaml_2.2.1       
 [9] pillar_1.5.1      lattice_0.20-41   glue_1.4.2        digest_0.6.27    
[13] promises_1.1.1    colorspace_2.0-0  htmltools_0.5.1.1 httpuv_1.5.4     
[17] Matrix_1.2-18     pkgconfig_2.0.3   purrr_0.3.4       scales_1.1.1     
[21] whisker_0.4       later_1.1.0.1     Rtsne_0.15        git2r_0.28.0     
[25] tibble_3.0.4      generics_0.1.0    farver_2.1.0      ellipsis_0.3.1   
[29] withr_2.4.1       cli_2.3.1         survival_3.2-7    magrittr_2.0.1   
[33] crayon_1.4.1      evaluate_0.14     ps_1.6.0          fs_1.5.0         
[37] fansi_0.4.2       MASS_7.3-53       tools_4.0.3       lifecycle_1.0.0  
[41] munsell_0.5.0     cluster_2.1.0     compiler_4.0.3    jquerylib_0.1.3  
[45] rsvd_1.0.3        rlang_0.4.10      grid_4.0.3        rstudioapi_0.13  
[49] htmlwidgets_1.5.3 labeling_0.4.2    rmarkdown_2.7     gtable_0.3.0     
[53] DBI_1.1.1         R6_2.5.0          knitr_1.31        utf8_1.1.4       
[57] rprojroot_2.0.2   stringi_1.5.3     Rcpp_1.0.5        vctrs_0.3.6      
[61] tidyselect_1.1.0  xfun_0.20        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
